---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    fig.path = "man/figures/README-",
    out.width = "100%"
)
```

# dnaEPICO

<!-- badges: start -->
[![GitHub issues](https://img.shields.io/github/issues/paulYRP/dnaEPICO)](https://github.com/paulYRP/dnaEPICO/issues)
[![GitHub pulls](https://img.shields.io/github/issues-pr/paulYRP/dnaEPICO)](https://github.com/paulYRP/dnaEPICO/pulls)
[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)
[![Bioc release status](http://www.bioconductor.org/shields/build/release/bioc/dnaEPICO.svg)](https://bioconductor.org/checkResults/release/bioc-LATEST/dnaEPICO)
[![Bioc devel status](http://www.bioconductor.org/shields/build/devel/bioc/dnaEPICO.svg)](https://bioconductor.org/checkResults/devel/bioc-LATEST/dnaEPICO)
<!-- [![Bioc downloads rank](https://bioconductor.org/shields/downloads/release/dnaEPICO.svg)](http://bioconductor.org/packages/stats/bioc/dnaEPICO/) -->
[![Bioc support](https://bioconductor.org/shields/posts/dnaEPICO.svg)](https://support.bioconductor.org/tag/dnaEPICO)
[![Bioc history](https://bioconductor.org/shields/years-in-bioc/dnaEPICO.svg)](https://bioconductor.org/packages/release/bioc/html/dnaEPICO.html#since)
[![Bioc last commit](https://bioconductor.org/shields/lastcommit/devel/bioc/dnaEPICO.svg)](http://bioconductor.org/checkResults/devel/bioc-LATEST/dnaEPICO/)
[![Bioc dependencies](https://bioconductor.org/shields/dependencies/release/dnaEPICO.svg)](https://bioconductor.org/packages/release/bioc/html/dnaEPICO.html#since)
[![Codecov test coverage](https://codecov.io/gh/paulYRP/dnapipeR/graph/badge.svg)](https://app.codecov.io/gh/paulYRP/dnapipeR)
<!-- badges: end -->

The goal of **`dnaEPICO`** is to provide a **modular, reproducible, and pipeline** for the preprocessing and statistical analysis of Illumina *EPICv2* DNA methylation array data.

The package integrates preprocessing, quality control, phenotype merging, generalised linear models (GLM), linear mixed-effects models (LME), and automated report generation. 
It is designed to run seamlessly on local machines as well as High-Performance Computing (HPC) environments via a **GNU Makeâ€“based workflow**.

## Installation instructions

Get the latest stable `R` release from [CRAN](http://cran.r-project.org/). Then install `dnaEPICO` from [Bioconductor](http://bioconductor.org/) using the following code:

```{r 'install', eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("dnaEPICO")
```

And the development version from [GitHub](https://github.com/paulYRP/dnaEPICO) with:

```{r 'install_dev', eval = FALSE}
BiocManager::install("paulYRP/dnaEPICO")
```

## Articles:
- [**A Pilot Epigenome-Wide Study of Posttraumatic Growth: Identifying Novel Candidates for Future Research**](https://www.mdpi.com/2075-4655/9/4/39)

## Tutorials:
- [**DNA Methylation Tutorial**](https://paulYRP.github.io/2025-cpgpneurogenomics-workshop/tutorial.html)
- [**Getting Started**](https://github.com/paulYRP/dnaEPICO/wiki/Getting-Started)
- [**Requirements**](https://github.com/paulYRP/dnaEPICO/wiki/Requirements)

## Citation

Below is the citation output from using `citation('dnaEPICO')` in R. Please
run this yourself to check for any updates on how to cite __dnaEPICO__.

```{r 'citation', eval = requireNamespace('dnaEPICO')}
print(citation('dnaEPICO'), bibtex = TRUE)
```

Please note that the `dnaEPICO` was only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package.

## Code of Conduct

Please note that the `dnaEPICO` project is released with a [Contributor Code of Conduct](http://bioconductor.org/about/code-of-conduct/). By contributing to this project, you agree to abide by its terms.

## Development tools

* Continuous code testing is possible thanks to [GitHub actions](https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/)  through `r BiocStyle::CRANpkg('usethis')`, `r BiocStyle::CRANpkg('remotes')`, and `r BiocStyle::CRANpkg('rcmdcheck')` customized to use [Bioconductor's docker containers](https://www.bioconductor.org/help/docker/) and `r BiocStyle::Biocpkg('BiocCheck')`.
* Code coverage assessment is possible thanks to [codecov](https://codecov.io/gh) and `r BiocStyle::CRANpkg('covr')`.
* The [documentation website](http://paulYRP.github.io/dnaEPICO) is automatically updated thanks to `r BiocStyle::CRANpkg('pkgdown')`.
* The code is styled automatically thanks to `r BiocStyle::CRANpkg('styler')`.
* The documentation is formatted thanks to `r BiocStyle::CRANpkg('devtools')` and `r BiocStyle::CRANpkg('roxygen2')`.

For more details, check the `dev` directory.

This package was developed using `r BiocStyle::Biocpkg('biocthis')`.
