<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Run preprocessingMinfiEwasWater.R as an external script from inst/script/.R Run preprocessingMinfiEwasWater as an external pipeline script — preprocessingMinfiEwasWater • dnaEPICO</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Run preprocessingMinfiEwasWater.R as an external script from inst/script/.R Run preprocessingMinfiEwasWater as an external pipeline script — preprocessingMinfiEwasWater"><meta property="og:description" content="Wrapper function that executes the preprocessingMinfiEwasWater.R script
shipped in inst/scripts using system calls."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dnaEPICO</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/dnaEPICO.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/paulYRP/dnaEPICO/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run preprocessingMinfiEwasWater.R as an external script from inst/script/.R Run preprocessingMinfiEwasWater as an external pipeline script</h1>
    <small class="dont-index">Source: <a href="https://github.com/paulYRP/dnaEPICO/blob/HEAD/R/preprocessingMinfiEwasWater.R" class="external-link"><code>R/preprocessingMinfiEwasWater.R</code></a></small>
    <div class="hidden name"><code>preprocessingMinfiEwasWater.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Wrapper function that executes the preprocessingMinfiEwasWater.R script
shipped in inst/scripts using system calls.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">preprocessingMinfiEwasWater</span><span class="op">(</span></span>
<span>  phenoFile <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater/pheno.csv"</span>,</span>
<span>  idatFolder <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater/idats"</span>,</span>
<span>  outputLogs <span class="op">=</span> <span class="st">"logs"</span>,</span>
<span>  nSamples <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  SampleID <span class="op">=</span> <span class="st">"Sample_Name"</span>,</span>
<span>  arrayType <span class="op">=</span> <span class="st">"IlluminaHumanMethylationEPICv2"</span>,</span>
<span>  annotationVersion <span class="op">=</span> <span class="st">"20a1.hg38"</span>,</span>
<span>  scriptLabel <span class="op">=</span> <span class="st">"preprocessingMinfiEwasWater"</span>,</span>
<span>  baseDataFolder <span class="op">=</span> <span class="st">"rData"</span>,</span>
<span>  sepType <span class="op">=</span> <span class="st">""</span>,</span>
<span>  tiffWidth <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>  tiffHeight <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  tiffRes <span class="op">=</span> <span class="fl">150</span>,</span>
<span>  qcCutoff <span class="op">=</span> <span class="fl">10.5</span>,</span>
<span>  detPtype <span class="op">=</span> <span class="st">"m+u"</span>,</span>
<span>  detPThreshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  funnormSeed <span class="op">=</span> <span class="fl">123</span>,</span>
<span>  normMethods <span class="op">=</span> <span class="st">"adjustedfunnorm"</span>,</span>
<span>  sexColumn <span class="op">=</span> <span class="st">"Sex"</span>,</span>
<span>  pvalThreshold <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  chrToRemove <span class="op">=</span> <span class="st">"chrX,chrY"</span>,</span>
<span>  snpsToRemove <span class="op">=</span> <span class="st">"SBE,CpG"</span>,</span>
<span>  mafThreshold <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  crossReactivePath <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater/12864_2024_10027_MOESM8_ESM.csv"</span>,</span>
<span>  plotGroupVar <span class="op">=</span> <span class="st">"Sex"</span>,</span>
<span>  lcRef <span class="op">=</span> <span class="st">"salivaEPIC"</span>,</span>
<span>  phenoOrder <span class="op">=</span> <span class="st">"Sample_Name;Timepoint;Sex;PredSex;Basename;Sentrix_ID;Sentrix_Position"</span>,</span>
<span>  lcPhenoDir <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-phenofile">phenoFile<a class="anchor" aria-label="anchor" href="#arg-phenofile"></a></dt>
<dd><p>Character. Path to phenotype CSV file.</p></dd>


<dt id="arg-idatfolder">idatFolder<a class="anchor" aria-label="anchor" href="#arg-idatfolder"></a></dt>
<dd><p>Character. Path to IDAT files directory.</p></dd>


<dt id="arg-outputlogs">outputLogs<a class="anchor" aria-label="anchor" href="#arg-outputlogs"></a></dt>
<dd><p>Character. Directory for log files.</p></dd>


<dt id="arg-nsamples">nSamples<a class="anchor" aria-label="anchor" href="#arg-nsamples"></a></dt>
<dd><p>Integer or NA. Number of samples to subset for testing.</p></dd>


<dt id="arg-sampleid">SampleID<a class="anchor" aria-label="anchor" href="#arg-sampleid"></a></dt>
<dd><p>Character. Sample identifier column name.</p></dd>


<dt id="arg-arraytype">arrayType<a class="anchor" aria-label="anchor" href="#arg-arraytype"></a></dt>
<dd><p>Character. Illumina array type.</p></dd>


<dt id="arg-annotationversion">annotationVersion<a class="anchor" aria-label="anchor" href="#arg-annotationversion"></a></dt>
<dd><p>Character. Annotation version.</p></dd>


<dt id="arg-scriptlabel">scriptLabel<a class="anchor" aria-label="anchor" href="#arg-scriptlabel"></a></dt>
<dd><p>Character. Label used in output naming.</p></dd>


<dt id="arg-basedatafolder">baseDataFolder<a class="anchor" aria-label="anchor" href="#arg-basedatafolder"></a></dt>
<dd><p>Character. Base directory for RData outputs.</p></dd>


<dt id="arg-septype">sepType<a class="anchor" aria-label="anchor" href="#arg-septype"></a></dt>
<dd><p>Character. Field separator for phenotype file.</p></dd>


<dt id="arg-tiffwidth">tiffWidth<a class="anchor" aria-label="anchor" href="#arg-tiffwidth"></a></dt>
<dd><p>Integer. Width of TIFF plots in pixels.</p></dd>


<dt id="arg-tiffheight">tiffHeight<a class="anchor" aria-label="anchor" href="#arg-tiffheight"></a></dt>
<dd><p>Integer. Height of TIFF plots in pixels.</p></dd>


<dt id="arg-tiffres">tiffRes<a class="anchor" aria-label="anchor" href="#arg-tiffres"></a></dt>
<dd><p>Integer. Resolution (DPI) for TIFF plots.</p></dd>


<dt id="arg-qccutoff">qcCutoff<a class="anchor" aria-label="anchor" href="#arg-qccutoff"></a></dt>
<dd><p>Numeric. Quality-control cutoff threshold.</p></dd>


<dt id="arg-detptype">detPtype<a class="anchor" aria-label="anchor" href="#arg-detptype"></a></dt>
<dd><p>Character. Detection p-value calculation type.</p></dd>


<dt id="arg-detpthreshold">detPThreshold<a class="anchor" aria-label="anchor" href="#arg-detpthreshold"></a></dt>
<dd><p>Numeric. Detection p-value threshold.</p></dd>


<dt id="arg-funnormseed">funnormSeed<a class="anchor" aria-label="anchor" href="#arg-funnormseed"></a></dt>
<dd><p>Integer. Random seed for normalization.</p></dd>


<dt id="arg-normmethods">normMethods<a class="anchor" aria-label="anchor" href="#arg-normmethods"></a></dt>
<dd><p>Character. Normalization method(s).</p></dd>


<dt id="arg-sexcolumn">sexColumn<a class="anchor" aria-label="anchor" href="#arg-sexcolumn"></a></dt>
<dd><p>Character. Sex column name in phenotype data.</p></dd>


<dt id="arg-pvalthreshold">pvalThreshold<a class="anchor" aria-label="anchor" href="#arg-pvalthreshold"></a></dt>
<dd><p>Numeric. Probe-level p-value threshold.</p></dd>


<dt id="arg-chrtoremove">chrToRemove<a class="anchor" aria-label="anchor" href="#arg-chrtoremove"></a></dt>
<dd><p>Character. Chromosomes to remove.</p></dd>


<dt id="arg-snpstoremove">snpsToRemove<a class="anchor" aria-label="anchor" href="#arg-snpstoremove"></a></dt>
<dd><p>Character. SNP probe types to remove.</p></dd>


<dt id="arg-mafthreshold">mafThreshold<a class="anchor" aria-label="anchor" href="#arg-mafthreshold"></a></dt>
<dd><p>Numeric. Minor allele frequency threshold.</p></dd>


<dt id="arg-crossreactivepath">crossReactivePath<a class="anchor" aria-label="anchor" href="#arg-crossreactivepath"></a></dt>
<dd><p>Character. Path to cross-reactive probe file.</p></dd>


<dt id="arg-plotgroupvar">plotGroupVar<a class="anchor" aria-label="anchor" href="#arg-plotgroupvar"></a></dt>
<dd><p>Character. Variable used for grouping plots.</p></dd>


<dt id="arg-lcref">lcRef<a class="anchor" aria-label="anchor" href="#arg-lcref"></a></dt>
<dd><p>Character. Reference panel for cell composition.</p></dd>


<dt id="arg-phenoorder">phenoOrder<a class="anchor" aria-label="anchor" href="#arg-phenoorder"></a></dt>
<dd><p>Character. Semicolon-separated phenotype column order.</p></dd>


<dt id="arg-lcphenodir">lcPhenoDir<a class="anchor" aria-label="anchor" href="#arg-lcphenodir"></a></dt>
<dd><p>Character. Output directory for cell composition phenotype.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Invisibly returns <code>NULL</code>. This function is called for its side effects,
performing Illumina EPICv2 preprocessing, quality control, normalisation,
probe filtering, cell composition estimation, and writing plots, logs,
and RData objects to disk.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.exists</a></span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="fu">preprocessingMinfiEwasWater</span><span class="op">(</span></span></span>
<span class="r-in"><span>  phenoFile <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater/pheno.csv"</span>,</span></span>
<span class="r-in"><span>  idatFolder <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater/idats"</span>,</span></span>
<span class="r-in"><span>  outputLogs <span class="op">=</span> <span class="st">"logs"</span>,</span></span>
<span class="r-in"><span>  nSamples <span class="op">=</span> <span class="cn">NA</span>,</span></span>
<span class="r-in"><span>  SampleID <span class="op">=</span> <span class="st">"Sample_Name"</span>,</span></span>
<span class="r-in"><span>  arrayType <span class="op">=</span> <span class="st">"IlluminaHumanMethylationEPICv2"</span>,</span></span>
<span class="r-in"><span>  annotationVersion <span class="op">=</span> <span class="st">"20a1.hg38"</span>,</span></span>
<span class="r-in"><span>  scriptLabel <span class="op">=</span> <span class="st">"preprocessingMinfiEwasWater"</span>,</span></span>
<span class="r-in"><span>  baseDataFolder <span class="op">=</span> <span class="st">"rData"</span>,</span></span>
<span class="r-in"><span>  sepType <span class="op">=</span> <span class="st">""</span>,</span></span>
<span class="r-in"><span>  tiffWidth <span class="op">=</span> <span class="fl">2000</span>,</span></span>
<span class="r-in"><span>  tiffHeight <span class="op">=</span> <span class="fl">1000</span>,</span></span>
<span class="r-in"><span>  tiffRes <span class="op">=</span> <span class="fl">150</span>,</span></span>
<span class="r-in"><span>  qcCutoff <span class="op">=</span> <span class="fl">10.5</span>,</span></span>
<span class="r-in"><span>  detPtype <span class="op">=</span> <span class="st">"m+u"</span>,</span></span>
<span class="r-in"><span>  detPThreshold <span class="op">=</span> <span class="fl">0.05</span>,</span></span>
<span class="r-in"><span>  funnormSeed <span class="op">=</span> <span class="fl">123</span>,</span></span>
<span class="r-in"><span>  normMethods <span class="op">=</span> <span class="st">"adjustedfunnorm"</span>,</span></span>
<span class="r-in"><span>  sexColumn <span class="op">=</span> <span class="st">"Sex"</span>,</span></span>
<span class="r-in"><span>  pvalThreshold <span class="op">=</span> <span class="fl">0.01</span>,</span></span>
<span class="r-in"><span>  chrToRemove <span class="op">=</span> <span class="st">"chrX,chrY"</span>,</span></span>
<span class="r-in"><span>  snpsToRemove <span class="op">=</span> <span class="st">"SBE,CpG"</span>,</span></span>
<span class="r-in"><span>  mafThreshold <span class="op">=</span> <span class="fl">0.1</span>,</span></span>
<span class="r-in"><span>  crossReactivePath <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater/12864_2024_10027_MOESM8_ESM.csv"</span>,</span></span>
<span class="r-in"><span>  plotGroupVar <span class="op">=</span> <span class="st">"Sex"</span>,</span></span>
<span class="r-in"><span>  lcRef <span class="op">=</span> <span class="st">"salivaEPIC"</span>,</span></span>
<span class="r-in"><span>  phenoOrder <span class="op">=</span> <span class="st">"Sample_Name;Timepoint;Sex;PredSex;Basename;Sentrix_ID;Sentrix_Position"</span>,</span></span>
<span class="r-in"><span>  lcPhenoDir <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Running preprocessingMinfiEwasWater:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Rscript "C:/Users/ruizpint/AppData/Local/Programs/R/R-4.4.1/library/dnaEPICO/scripts/preprocessingMinfiEwasWater.R" --phenoFile "data/preprocessingMinfiEwasWater/pheno.csv" --sepType "" --idatFolder "data/preprocessingMinfiEwasWater/idats" --outputLogs "logs" --nSamples NA --SampleID "Sample_Name" --arrayType "IlluminaHumanMethylationEPICv2" --annotationVersion "20a1.hg38" --scriptLabel "preprocessingMinfiEwasWater" --baseDataFolder "rData" --tiffWidth 2000 --tiffHeight 1000 --tiffRes 150 --qcCutoff 10.5 --detPtype "m+u" --detPThreshold 0.05 --funnormSeed 123 --normMethods "adjustedfunnorm" --sexColumn "Sex" --pvalThreshold 0.01 --chrToRemove "chrX,chrY" --snpsToRemove "SBE,CpG" --mafThreshold 0.1 --crossReactivePath "data/preprocessingMinfiEwasWater/12864_2024_10027_MOESM8_ESM.csv" --plotGroupVar "Sex" --lcRef "salivaEPIC" --phenoOrder "Sample_Name;Timepoint;Sex;PredSex;Basename;Sentrix_ID;Sentrix_Position" --lcPhenoDir "data/preprocessingMinfiEwasWater"</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'Rscript "C:/Users/ruizpint/AppData/Local/Programs/R/R-4.4.1/library/dnaEPICO/scripts/preprocessingMinfiEwasWater.R" --phenoFile "data/preprocessingMinfiEwasWater/pheno.csv" --sepType "" --idatFolder "data/preprocessingMinfiEwasWater/idats" --outputLogs "logs" --nSamples NA --SampleID "Sample_Name" --arrayType "IlluminaHumanMethylationEPICv2" --annotationVersion "20a1.hg38" --scriptLabel "preprocessingMinfiEwasWater" --baseDataFolder "rData" --tiffWidth 2000 --tiffHeight 1000 --tiffRes 150 --qcCutoff 10.5 --detPtype "m+u" --detPThreshold 0.05 --funnormSeed 123 --normMethods "adjustedfunnorm" --sexColumn "Sex" --pvalThreshold 0.01 --chrToRemove "chrX,chrY" --snpsToRemove "SBE,CpG" --mafThreshold 0.1 --crossReactivePath "data/preprocessingMinfiEwasWater/12864_2024_10027_MOESM8_ESM.csv" --plotGroupVar "Sex" --lcRef "salivaEPIC" --phenoOrder "Sample_Name;Timepoint;Sex;PredSex;Basename;Sentrix_ID;Sentrix_Position" --lcPhenoDir "data/preprocessingMinfiEwasWater"' execution failed with error code 1</span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Paul Ruiz, Divya Metha.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer></div>






  </body></html>

