<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>The dnaEPICO User's Guide • dnaEPICO</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="The dnaEPICO User's Guide">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dnaEPICO</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/dnaEPICO.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/paulYRP/dnaEPICO/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>The dnaEPICO User’s Guide</h1>
                        <h4 data-toc-skip class="author">Paul Ruiz</h4>
            <address class="author_afil">
      Queensland University of
Technology<br><a class="author_email" href="mailto:#"></a><a href="mailto:ruizpint@qut.edu.au" class="email">ruizpint@qut.edu.au</a>
      </address>
                              <h4 data-toc-skip class="author">Divya
Metha</h4>
            <address class="author_afil">
      Queensland University of Technology<br><h4 data-toc-skip class="date">2026-01-23</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/paulYRP/dnaEPICO/blob/HEAD/vignettes/dnaEPICO.Rmd" class="external-link"><code>vignettes/dnaEPICO.Rmd</code></a></small>
      <div class="hidden name"><code>dnaEPICO.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>A comprehensive guide to using the dnaEPICO package. The
      package supports preprocessing and analysis of Illumina EPICv2 DNA
      methylation array data, including QC, filtering, SVA, cell-type
      deconvolution, phenotype integration, modelling, and export-ready
      files for the DNAmAge Clock Foundation platform.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <strong>dnaEPICO</strong> package supports preprocessing and
analysis of Illumina EPICv2 DNA methylation array data. It covers data
import, quality control, surrogate variable analysis, probe and sample
filtering, cell-type deconvolution, phenotype integration, and
downstream modelling using generalised linear and linear mixed-effects
models. The package integrates with Bioconductor workflows and tools
including <code>minfi</code>, <code>ENmix</code>, and
<code>wateRmelon</code>. Additional functionality is provided via the
<code>ewastools</code> package, which is installed from GitHub using
<code>devtools::install_github("hhhh5/ewastools")</code>.</p>
<p>dnaEPICO also prepares standardised DNA methylation files for direct
export to the <a href="https://dnamage.clockfoundation.org/user" class="external-link">DNAmAge
Clock Foundation platform</a>.</p>
<div class="section level3">
<h3 id="citing-dnaepico">Citing <code>dnaEPICO</code><a class="anchor" aria-label="anchor" href="#citing-dnaepico"></a>
</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.20/dnaEPICO" class="external-link">dnaEPICO</a></em>
package implements methods described across multiple independent
publications by different authors. As a result, no single manuscript
fully describes all components of the package. This section provides
guidance on how to cite dnaEPICO appropriately.</p>
<p>The citation guidance below is adapted from the user guides and
vignettes of the packages on which dnaEPICO builds, <a href="https://github.com/hansenlab/minfi/blob/devel/vignettes/minfi.Rmd" class="external-link">minfi</a>,
<a href="https://github.com/xuz1/ENmix/blob/master/vignettes/ENmix.Rmd" class="external-link">ENmix</a>,
<a href="https://github.com/schalkwyk/wateRmelon/blob/master/vignettes/references.bib" class="external-link">wateRmelon</a>,
and <a href="https://hhhh5.github.io/ewastools/articles/exemplary_ewas.html" class="external-link">ewastools</a>.</p>
<ul>
<li>If you use <code><a href="../reference/preprocessingMinfiEwasWater.html">preprocessingMinfiEwasWater()</a></code>, please cite
<span class="citation">(Aryee et al. 2014)</span>, <span class="citation">(Fortin, Triche Jr., and Hansen 2017)</span>, <span class="citation">(Xu, Niu, and Taylor 2021)</span>, <span class="citation">(Pidsley et al. 2013)</span>, <span class="citation">(Maksimovic, Gordon, and Oshlack 2012)</span>, <span class="citation">(Fortin et al. 2014)</span>, <span class="citation">(Triche et al. 2013)</span> and <span class="citation">(Touleimat and Tost 2012)</span>.</li>
<li>If you use <code><a href="../reference/svaEnmix.html">svaEnmix()</a></code>, please cite <span class="citation">(Aryee et al. 2014)</span>, <span class="citation">(Xu,
Niu, and Taylor 2021)</span>, and <span class="citation">(Murat et al.
2020)</span>.</li>
<li>If you use <code>make f3 MODEL=model1</code>, please cite <span class="citation">(Aryee et al. 2014)</span>, <span class="citation">(Fortin, Triche Jr., and Hansen 2017)</span>, <span class="citation">(Xu, Niu, and Taylor 2021)</span>, <span class="citation">(Pidsley et al. 2013)</span>, <span class="citation">(Maksimovic, Gordon, and Oshlack 2012)</span>, <span class="citation">(Fortin et al. 2014)</span>, <span class="citation">(Triche et al. 2013)</span>, <span class="citation">(Touleimat and Tost 2012)</span> and <span class="citation">(Murat et al. 2020)</span>.</li>
<li>If you use <code>make f4 MODEL=model1</code>, please cite <span class="citation">(Aryee et al. 2014)</span>, <span class="citation">(Fortin, Triche Jr., and Hansen 2017)</span>, <span class="citation">(Xu, Niu, and Taylor 2021)</span>, <span class="citation">(Pidsley et al. 2013)</span>, <span class="citation">(Maksimovic, Gordon, and Oshlack 2012)</span>, <span class="citation">(Fortin et al. 2014)</span>, <span class="citation">(Triche et al. 2013)</span>, <span class="citation">(Touleimat and Tost 2012)</span>, <span class="citation">(Murat et al. 2020)</span> and <span class="citation">(Marschner 2011)</span>.</li>
<li>If you use <code>make f3lme MODEL=model1</code>, please cite <span class="citation">(Aryee et al. 2014)</span>, <span class="citation">(Fortin, Triche Jr., and Hansen 2017)</span>, <span class="citation">(Xu, Niu, and Taylor 2021)</span>, <span class="citation">(Pidsley et al. 2013)</span>, <span class="citation">(Maksimovic, Gordon, and Oshlack 2012)</span>, <span class="citation">(Fortin et al. 2014)</span>, <span class="citation">(Triche et al. 2013)</span>, <span class="citation">(Touleimat and Tost 2012)</span>, <span class="citation">(Murat et al. 2020)</span> and <span class="citation">(Bates et al. 2015)</span>.</li>
<li>If you use <code>make all MODEL=model1</code>, please cite <span class="citation">(Aryee et al. 2014)</span>, <span class="citation">(Fortin, Triche Jr., and Hansen 2017)</span>, <span class="citation">(Xu, Niu, and Taylor 2021)</span>, <span class="citation">(Pidsley et al. 2013)</span>, <span class="citation">(Maksimovic, Gordon, and Oshlack 2012)</span>, <span class="citation">(Fortin et al. 2014)</span>, <span class="citation">(Triche et al. 2013)</span>, <span class="citation">(Touleimat and Tost 2012)</span>, <span class="citation">(Murat et al. 2020)</span>, <span class="citation">(Marschner 2011)</span> and <span class="citation">(Bates et al. 2015)</span>.</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.20/dnaEPICO" class="external-link">dnaEPICO</a></em>
implements methods in <em><a href="https://bioconductor.org/packages/3.20/minfi" class="external-link">minfi</a></em>,
<em><a href="https://bioconductor.org/packages/3.20/ENmix" class="external-link">ENmix</a></em> and
<em><a href="https://bioconductor.org/packages/3.20/wateRmelon" class="external-link">wateRmelon</a></em>
packages so there is quite the number of dependencies which need to be
installed. Additional functionality is provided via the
<code>ewastools</code> package. This can be handled by simply
running:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"dnaEPICO"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"hhhh5/ewastools"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/paulYRP/dnaEPICO" class="external-link">"dnaEPICO"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://hhhh5.github.io/ewastools/" class="external-link">"ewastools"</a></span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, <em><a href="https://bioconductor.org/packages/3.20/dnaEPICO" class="external-link">dnaEPICO</a></em>
can be installed directly from our <a href="https://github.com/paulYRP/dnaEPICO" class="external-link">github</a></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'devtools'</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'paulYRP/dnaEPICO'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.20/dnaEPICO" class="external-link">dnaEPICO</a></em>
is a DNA methylation preprocessing and modelling pipeline designed for
Illumina <strong>EPICv2</strong> arrays. It wraps reproducible analysis
steps into a consistent structure that:</p>
<ul>
<li>reads <strong>IDATs</strong> and phenotype metadata,</li>
<li>performs <strong>QC</strong>, filtering and normalisation,</li>
<li>computes surrogate variables (<strong>SVA</strong>),</li>
<li>
<strong>merges phenotype</strong> with methylation matrices,</li>
<li>runs <strong>GLM</strong> (cross-sectional) and
<strong>GLMM/LME</strong> (longitudinal) models,</li>
<li>writes <strong>logs</strong>, <strong>figures</strong>,
<strong>RData objects</strong>, and a <strong>PDF report</strong>.</li>
</ul>
<p>A core design principle is that each step creates its own folders and
outputs, so you can run steps independently or chain them through the
Makefile pipeline.</p>
<div class="section level3">
<h3 id="localcluster-use">Local/Cluster use<a class="anchor" aria-label="anchor" href="#localcluster-use"></a>
</h3>
<p><strong>Local use</strong> is recommended for:</p>
<ul>
<li>testing the pipeline end-to-end on a small subset,</li>
<li>preparing clean merged data objects,</li>
<li>generating QC plots and logs,</li>
<li>validating input formatting and parameter choices.</li>
</ul>
<p><strong>Cluster/HPC use</strong> is recommended for:</p>
<ul>
<li>genome-wide modelling (hundreds of thousands of CpGs),</li>
<li>parallel chunk processing,</li>
<li>large memory operations and multi-core execution,</li>
<li>reproducible batch execution with Make targets (<code>f3</code>,
<code>f4</code>, <code>f3lme</code> and <code>all</code>).</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="data-expectations-and-folder-conventions">Data expectations and folder conventions<a class="anchor" aria-label="anchor" href="#data-expectations-and-folder-conventions"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.20/dnaEPICO" class="external-link">dnaEPICO</a></em>
expects:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">├──</span> data/                                   <span class="co"># Processed data outputs (per model)</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">│</span>   ├── preprocessingMinfiEwasWater/</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="ex">│</span>   │   ├── idats/                          <span class="co"># Folder containing raw IDAT files</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="ex">│</span>   │   ├── pheno.csv                       <span class="co"># Phenotype file with sample metadata</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="ex">│</span>   │   └── 12864_2024_10027_MOESM8_ESM.csv <span class="co"># Cross-reactivity comparison reference</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="ex">│</span>   ├── Makefile                            <span class="co"># Arguments </span></span></code></pre></div>
<ul>
<li>
<strong>Phenotype file</strong>: a CSV with sample identifiers and
covariates.</li>
<li>
<strong>IDAT folder</strong>: contains Illumina
<code>*_Red.idat</code> and <code>*_Grn.idat</code> pairs.</li>
<li>
<strong>Output folders</strong>: created by each step (logs, rData,
figures, reports).</li>
</ul>
<p>You can download the list of off-target cross-hybridisation sites for
EPICv2 probes from <a href="https://springernature.figshare.com/articles/dataset/Additional_file_8_of_Characterisation_and_reproducibility_of_the_HumanMethylationEPIC_v2_0_BeadChip_for_DNA_methylation_profiling/26689706?file=48531323" class="external-link"><code>12864_2024_10027_MOESM8_ESM.csv</code></a>.</p>
<p>You can extract the <code>Makefile</code> using:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/extractMake.html">extractMake</a></span><span class="op">(</span></span>
<span>  destDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><em><a href="https://bioconductor.org/packages/3.20/dnaEPICO" class="external-link">dnaEPICO</a></em>
generates:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">├──</span> data/                                   <span class="co"># Processed data outputs (per model)</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="ex">│</span>   ├── preprocessingMinfiEwasWater/</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="ex">│</span>   │   ├── idats/                          <span class="co"># Folder containing raw IDAT files</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="ex">│</span>   │   ├── pheno.csv                       <span class="co"># Phenotype file with sample metadata</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="ex">│</span>   │   └── 12864_2024_10027_MOESM8_ESM.csv <span class="co"># Cross-reactivity comparison reference</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="ex">│</span>   ├── model1/                             <span class="co"># Outputs specific to the first model</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="ex">│</span>   │   ├── preprocessingMinfiEwasWater/    <span class="co"># Preprocessed phenotype + QC from Minfi</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="ex">│</span>   │   ├── svaEnmix/                       <span class="co"># Surrogate variable analysis </span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="ex">│</span>   │   ├── preprocessingPheno/             <span class="co"># Cleaned and merged phenotype files </span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="ex">│</span>   │   ├── methylationGLM_T1/              <span class="co"># CpG-level GLM results (single timepoint)</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="ex">│</span>   │   ├── methylationGLMM_T1T2/           <span class="co"># Longitudinal LME results (T1 vs T2)</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a><span class="ex">│</span>   │   └── ...                             <span class="co"># Additional subfolders by model</span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a><span class="ex">│</span>   ├── model2/</span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a><span class="ex">│</span>   └── model3/</span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a><span class="ex">├──</span> rData/                                  <span class="co"># Processed R objects (MSet, Beta, CN matrices, etc.)</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="ex">├──</span> logs/                                   <span class="co"># Logging output from script runs</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="ex">├──</span> figures/                                <span class="co"># QC plots and visualization output</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a><span class="ex">├──</span> results/                                <span class="co"># Output tables and summary statistics</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a><span class="ex">├──</span> reports/                                <span class="co"># Optional rendered reports (PDF)</span></span></code></pre></div>
<p>Each step writes:</p>
<ul>
<li>
<code>logs/</code>: logs</li>
<li>
<code>rData/</code>: intermediate and final R objects</li>
<li>
<code>figures/</code>: plots organised by step name</li>
<li>
<code>data/</code>: exported tables (annotated results, phenotype
exports)</li>
<li>
<code>reports/</code>: final PDF report</li>
</ul>
<p>This <em>“each function writes to its own folder”</em> approach is
intentional: it improves reproducibility, debugging, and HPC
resuming.</p>
</div>
<div class="section level2">
<h2 id="local-use">Local use<a class="anchor" aria-label="anchor" href="#local-use"></a>
</h2>
<p>Local use focuses on the first three steps that are commonly run
interactively:</p>
<ol style="list-style-type: decimal">
<li>
<code><a href="../reference/preprocessingMinfiEwasWater.html">preprocessingMinfiEwasWater()</a></code>: import + QC + normalise
+ filter + cell composition</li>
<li>
<code><a href="../reference/svaEnmix.html">svaEnmix()</a></code>: surrogate variable analysis using ENmix
control probes</li>
<li>
<code><a href="../reference/preprocessingPheno.html">preprocessingPheno()</a></code>: merge phenotype metadata with
methylation matrices</li>
</ol>
<div class="section level3">
<h3 id="step-1-preprocessingminfiewaswater">Step 1: <code>preprocessingMinfiEwasWater()</code><a class="anchor" aria-label="anchor" href="#step-1-preprocessingminfiewaswater"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/preprocessingMinfiEwasWater.html">preprocessingMinfiEwasWater</a></span><span class="op">(</span></span>
<span>    phenoFile <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater/pheno.csv"</span>,</span>
<span>    idatFolder <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater/idats"</span>,</span>
<span>    outputLogs <span class="op">=</span> <span class="st">"logs"</span>,</span>
<span>    nSamples <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>    SampleID <span class="op">=</span> <span class="st">"Sample_Name"</span>,</span>
<span>    arrayType <span class="op">=</span> <span class="st">"IlluminaHumanMethylationEPICv2"</span>,</span>
<span>    annotationVersion <span class="op">=</span> <span class="st">"20a1.hg38"</span>,</span>
<span>    scriptLabel <span class="op">=</span> <span class="st">"preprocessingMinfiEwasWater"</span>,</span>
<span>    baseDataFolder <span class="op">=</span> <span class="st">"rData"</span>,</span>
<span>    sepType <span class="op">=</span> <span class="st">""</span>,</span>
<span>    tiffWidth <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>    tiffHeight <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>    tiffRes <span class="op">=</span> <span class="fl">150</span>,</span>
<span>    qcCutoff <span class="op">=</span> <span class="fl">10.5</span>,</span>
<span>    detPtype <span class="op">=</span> <span class="st">"m+u"</span>,</span>
<span>    detPThreshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    funnormSeed <span class="op">=</span> <span class="fl">123</span>,</span>
<span>    normMethods <span class="op">=</span> <span class="st">"adjustedfunnorm"</span>,</span>
<span>    sexColumn <span class="op">=</span> <span class="st">"Sex"</span>,</span>
<span>    pvalThreshold <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>    chrToRemove <span class="op">=</span> <span class="st">"chrX,chrY"</span>,</span>
<span>    snpsToRemove <span class="op">=</span> <span class="st">"SBE,CpG"</span>,</span>
<span>    mafThreshold <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>    crossReactivePath <span class="op">=</span></span>
<span>      <span class="st">"data/preprocessingMinfiEwasWater/12864_2024_10027_MOESM8_ESM.csv"</span>,</span>
<span>    plotGroupVar <span class="op">=</span> <span class="st">"Sex"</span>,</span>
<span>    lcRef <span class="op">=</span> <span class="st">"salivaEPIC"</span>,</span>
<span>    phenoOrder <span class="op">=</span> <span class="st">"Sample_Name;Timepoint;Sex;PredSex;Basename;Sentrix_ID;Sentrix_Position"</span>,</span>
<span>    lcPhenoDir <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="what-it-does">What it does:<a class="anchor" aria-label="anchor" href="#what-it-does"></a>
</h4>
<p>This wrapper executes the pipeline script in
<code>inst/scripts/preprocessingMinfiEwasWater.R</code> using
<code>Rscript</code>. Conceptually, it:</p>
<ol style="list-style-type: decimal">
<li>Reads phenotype CSV (<code>phenoFile</code>) and IDATs
(<code>idatFolder</code>).</li>
<li>Loads an <code>RGChannelSet</code> (minfi) and runs basic QC.</li>
<li>Creates
<code>MSet</code>/<code>RatioSet</code>/<code>GenomicRatioSet</code>
objects.</li>
<li>Computes detection p-values and filters failed probes/samples.</li>
<li>Applies normalisation (<code>normMethods</code>).</li>
<li>Performs probe filtering (p-value, chrX/Y removal, SNP filters,
cross-reactive probes).</li>
<li>Exports final matrices (Beta/M/CN) to <code>rData/</code>.</li>
<li>Produces plots in <code>figures/</code> and logs in
<code>logs/</code>.</li>
<li>Optionally estimates cell composition (via <code>lcRef</code>) and
writes <code>phenoLC.csv</code> into <code>lcPhenoDir</code>.</li>
</ol>
</div>
<div class="section level4">
<h4 id="parameter-guide">Parameter guide:<a class="anchor" aria-label="anchor" href="#parameter-guide"></a>
</h4>
<ul>
<li>
<code>phenoFile</code>: where your sample metadata lives (CSV).</li>
<li>
<code>idatFolder</code>: directory with IDAT pairs.</li>
<li>
<code>nSamples</code>: subset for testing locally
(e.g. <code>5</code>).</li>
<li>
<code>SampleID</code>: name of the sample ID column in your
phenotype file.</li>
<li>
<code>arrayType</code>, <code>annotationVersion</code>: ensures the
correct EPICv2 manifest/annotation is used.</li>
<li>
<code>qcCutoff</code>, <code>detPtype</code>,
<code>detPThreshold</code>: QC rules; detection p-value is used to flag
failed probes/samples.</li>
<li>
<code>normMethods</code>: normalisation method(s) you want to
apply.</li>
<li>
<code>pvalThreshold</code>, <code>chrToRemove</code>,
<code>snpsToRemove</code>, <code>mafThreshold</code>: probe-level
filters.</li>
<li>
<code>crossReactivePath</code>: CSV containing cross-reactive probes
to remove.</li>
<li>
<code>phenoOrder</code>: a semicolon-delimited string that defines
how the final phenotype table is ordered.</li>
<li>
<code>lcRef</code>, <code>lcPhenoDir</code>: cell composition
reference + output for pheno with cell fractions.</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="step-2-svaenmix">Step 2: <code>svaEnmix()</code><a class="anchor" aria-label="anchor" href="#step-2-svaenmix"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/svaEnmix.html">svaEnmix</a></span><span class="op">(</span></span>
<span>    phenoFile <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater/phenoLC.csv"</span>,</span>
<span>    rgsetData <span class="op">=</span> <span class="st">"rData/preprocessingMinfiEwasWater/objects/RGSet.RData"</span>,</span>
<span>    sepType <span class="op">=</span> <span class="st">""</span>,</span>
<span>    outputLogs <span class="op">=</span> <span class="st">"logs"</span>,</span>
<span>    nSamples <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>    SampleID <span class="op">=</span> <span class="st">"Sample_Name"</span>,</span>
<span>    arrayType <span class="op">=</span> <span class="st">"IlluminaHumanMethylationEPICv2"</span>,</span>
<span>    annotationVersion <span class="op">=</span> <span class="st">"20a1.hg38"</span>,</span>
<span>    SentrixIDColumn <span class="op">=</span> <span class="st">"Sentrix_ID"</span>,</span>
<span>    SentrixPositionColumn <span class="op">=</span> <span class="st">"Sentrix_Position"</span>,</span>
<span>    ctrlSvaPercVar <span class="op">=</span> <span class="fl">0.90</span>,</span>
<span>    ctrlSvaFlag <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    scriptLabel <span class="op">=</span> <span class="st">"svaEnmix"</span>,</span>
<span>    tiffWidth <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>    tiffHeight <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>    tiffRes <span class="op">=</span> <span class="fl">150</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="what-it-does-1">What it does:<a class="anchor" aria-label="anchor" href="#what-it-does-1"></a>
</h4>
<p>This wrapper executes <code>inst/scripts/svaEnmix.R</code>.
Conceptually, it:</p>
<ol style="list-style-type: decimal">
<li>Loads <code>RGSet</code> from <code>rgsetData</code> and phenotype
from <code>phenoFile</code>.</li>
<li>Uses ENmix control probes to derive control-based components.</li>
<li>Chooses how many components to keep based on
<code>ctrlSvaPercVar</code>.</li>
<li>Writes summary outputs, figures, and updated objects/logs.</li>
</ol>
</div>
<div class="section level4">
<h4 id="parameter-guide-1">Parameter guide:<a class="anchor" aria-label="anchor" href="#parameter-guide-1"></a>
</h4>
<ul>
<li>
<code>rgsetData</code>: the saved object from Step 1 (input to
SVA).</li>
<li>
<code>phenoFile</code>: <code>phenoLC.csv</code> from Step 1
(includes cell fractions).</li>
<li>
<code>ctrlSvaPercVar</code>: proportion of variance to capture using
control-derived SVs.</li>
<li>
<code>SentrixIDColumn</code>, <code>SentrixPositionColumn</code>:
needed to match the array sample structure.</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="step-3-preprocessingpheno">Step 3: <code>preprocessingPheno()</code><a class="anchor" aria-label="anchor" href="#step-3-preprocessingpheno"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/preprocessingPheno.html">preprocessingPheno</a></span><span class="op">(</span></span>
<span>    phenoFile <span class="op">=</span> <span class="st">"data/preprocessingMinfiEwasWater/phenoLC.csv"</span>,</span>
<span>    sepType <span class="op">=</span> <span class="st">""</span>,</span>
<span>    betaPath <span class="op">=</span></span>
<span>      <span class="st">"rData/preprocessingMinfiEwasWater/metrics/beta_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData"</span>,</span>
<span>    mPath <span class="op">=</span></span>
<span>      <span class="st">"rData/preprocessingMinfiEwasWater/metrics/m_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData"</span>,</span>
<span>    cnPath <span class="op">=</span></span>
<span>      <span class="st">"rData/preprocessingMinfiEwasWater/metrics/cn_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData"</span>,</span>
<span>    SampleID <span class="op">=</span> <span class="st">"Sample_Name"</span>,</span>
<span>    timeVar <span class="op">=</span> <span class="st">"Timepoint"</span>,</span>
<span>    timepoints <span class="op">=</span> <span class="st">"1,2"</span>,</span>
<span>    combineTimepoints <span class="op">=</span> <span class="st">"1,2"</span>,</span>
<span>    outputPheno <span class="op">=</span> <span class="st">"data/preprocessingPheno"</span>,</span>
<span>    outputRData <span class="op">=</span> <span class="st">"rData/preprocessingPheno/metrics"</span>,</span>
<span>    outputRDataMerge <span class="op">=</span> <span class="st">"rData/preprocessingPheno/mergeData"</span>,</span>
<span>    sexColumn <span class="op">=</span> <span class="st">"Sex"</span>,</span>
<span>    outputLogs <span class="op">=</span> <span class="st">"logs"</span>,</span>
<span>    outputDir <span class="op">=</span> <span class="st">"data/preprocessingPheno"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="what-it-does-2">What it does:<a class="anchor" aria-label="anchor" href="#what-it-does-2"></a>
</h4>
<p>This wrapper executes <code>inst/scripts/preprocessingPheno.R</code>.
Conceptually, it:</p>
<ol style="list-style-type: decimal">
<li><p>Loads phenotype and methylation matrices (Beta/M/CN).</p></li>
<li><p>Aligns samples by <code>SampleID</code>.</p></li>
<li><p>Creates timepoint-specific subsets
(<code>timepoints</code>).</p></li>
<li><p>Optionally combines timepoints (<code>combineTimepoints</code>)
for longitudinal sets.</p></li>
<li>
<p>Writes:</p>
<ul>
<li>exported phenotype tables into <code>outputPheno</code>,</li>
<li>methylation metric objects into <code>outputRData</code>,</li>
<li>merged objects (e.g., <code>phenoBetaT1.RData</code>) into
<code>outputRDataMerge</code>.</li>
</ul>
</li>
</ol>
</div>
<div class="section level4">
<h4 id="parameter-guide-2">Parameter guide:<a class="anchor" aria-label="anchor" href="#parameter-guide-2"></a>
</h4>
<ul>
<li>
<code>phenoFile</code>: <code>phenoLC.csv</code> generated in Step
1.</li>
<li>
<code>betaPath</code>, <code>mPath</code>, <code>cnPath</code>:
paths to saved methylation metric objects (Beta, M, and CN matrices)
produced during preprocessing.</li>
<li>
<code>SampleID</code>: column name in the phenotype file used to
match samples to methylation matrices.</li>
<li>
<code>timeVar</code>: phenotype column that defines the time
variable (e.g. baseline, follow-up).</li>
<li>
<code>timepoints</code>: comma-separated list of timepoints to
subset and export individually.</li>
<li>
<code>combineTimepoints</code>: comma-separated list of timepoints
to merge into a single longitudinal dataset.</li>
<li>
<code>outputPheno</code>: directory where cleaned and subsetted
phenotype tables are written.</li>
<li>
<code>outputRData</code>: directory where timepoint-specific
methylation metric objects are saved.</li>
<li>
<code>outputRDataMerge</code>: directory where merged
phenotype–methylation objects are written.</li>
<li>
<code>sexColumn</code>: column used to store biological sex.</li>
<li>
<code>outputLogs</code>: directory where execution logs are
written.</li>
<li>
<code>outputDir</code>: base directory for exported data products
generated by this step.</li>
</ul>
<p>This step prepares analysis-ready phenotype–methylation objects for
downstream GLM and LME modelling and generates files compatible with the
DNAmAge Clock Foundation platform (see your
<code>log_preprocessingPheno.txt</code> for details).</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="cluster-hpc-use">Cluster / HPC use<a class="anchor" aria-label="anchor" href="#cluster-hpc-use"></a>
</h2>
<p>On full datasets, model fitting across hundreds of thousands of CpGs
typically requires:</p>
<ul>
<li>large memory,</li>
<li>parallel chunking,</li>
<li>long runtimes.</li>
</ul>
<p><code>dnaEPICO</code> supports this with a Makefile-driven pipeline
that can run the pipeline end-to-end or by grouped targets.</p>
<div class="section level3">
<h3 id="step-4-methylationglm_t1">Step 4: <code>methylationGLM_T1()</code><a class="anchor" aria-label="anchor" href="#step-4-methylationglm_t1"></a>
</h3>
<p>This step is typically executed on <strong>HPC systems</strong> due
to memory and CPU requirements.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/methylationGLM_T1.html">methylationGLM_T1</a></span><span class="op">(</span></span>
<span>    inputPheno <span class="op">=</span> <span class="st">"rData/preprocessingPheno/mergeData/phenoBetaT1.RData"</span>,</span>
<span>    outputLogs <span class="op">=</span> <span class="st">"logs"</span>,</span>
<span>    outputRData <span class="op">=</span> <span class="st">"rData/methylationGLM_T1/models"</span>,</span>
<span>    outputPlots <span class="op">=</span> <span class="st">"figures/methylationGLM_T1"</span>,</span>
<span>    phenotypes <span class="op">=</span> <span class="st">"Pheno1,Pheno2,Pheno3,Pheno4"</span>,</span>
<span>    covariates <span class="op">=</span> <span class="st">"Sex,Age,Ethnicity,TraumaDefinition,Leukocytes,Epithelial.cells"</span>,</span>
<span>    factorVars <span class="op">=</span> <span class="st">"Sex,Ethnicity,TraumaDefinition"</span>,</span>
<span>    cpgPrefix <span class="op">=</span> <span class="st">"cg"</span>,</span>
<span>    cpgLimit <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>    nCores <span class="op">=</span> <span class="fl">32</span>,</span>
<span>    plotWidth <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>    plotHeight <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>    plotDPI <span class="op">=</span> <span class="fl">150</span>,</span>
<span>    interactionTerm <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    libPath <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    glmLibs <span class="op">=</span> <span class="st">"glm2"</span>,</span>
<span>    prsMap <span class="op">=</span> <span class="st">"Pheno1:PRS_1,Pheno2:PRS_2,Pheno3:PRS_3,Pheno4:PRS_4"</span>,</span>
<span>    summaryPval <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>    summaryResidualSD <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    saveSignificantCpGs <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    significantCpGDir <span class="op">=</span> <span class="st">"preliminaryResults/cpgs/methylationGLM_T1"</span>,</span>
<span>    significantCpGPval <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    saveTxtSummaries <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    chunkSize <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>    summaryTxtDir <span class="op">=</span> <span class="st">"preliminaryResults/summary/methylationGLM_T1/glm"</span>,</span>
<span>    fdrThreshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    padjmethod <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>    annotationPackage <span class="op">=</span> <span class="st">"IlluminaHumanMethylationEPICv2anno.20a1.hg38"</span>,</span>
<span>    annotationCols <span class="op">=</span> <span class="st">"Name,chr,pos,UCSC_RefGene_Group,UCSC_RefGene_Name,</span></span>
<span><span class="st">                      Relation_to_Island,GencodeV41_Group"</span>,</span>
<span>    annotatedGLMOut <span class="op">=</span> <span class="st">"data/methylationGLM_T1"</span></span>
<span><span class="op">)</span> </span></code></pre></div>
<div class="section level4">
<h4 id="what-it-does-3">What it does:<a class="anchor" aria-label="anchor" href="#what-it-does-3"></a>
</h4>
<p>This wrapper executes the external script
<code>inst/scripts/methylationGLM_T1.R</code> using a system call.
Conceptually, it:</p>
<ol style="list-style-type: decimal">
<li><p>Loads a merged phenotype–beta-value object
(<code>inputPheno</code>) produced by
<code><a href="../reference/preprocessingPheno.html">preprocessingPheno()</a></code>.</p></li>
<li><p>Iterates over CpGs in chunks and fits a generalised linear model
<a href="https://cran.r-project.org/web/packages/glm2/index.html" class="external-link">(glm2)</a>
for each CpG.</p></li>
<li><p>Models each specified phenotype separately, adjusting for
covariates and factor variables.</p></li>
<li><p>Optionally includes interaction terms and polygenic risk score
(PRS) mappings.</p></li>
<li><p>Applies multiple-testing correction (FDR).</p></li>
<li>
<p>Writes:</p>
<ul>
<li>fitted model objects to <code>outputRData</code>,</li>
<li>diagnostic and summary plots to <code>outputPlots</code>,</li>
<li>log files to <code>outputLogs</code>,</li>
<li>text summaries and significant CpG tables,</li>
<li>annotated result tables using the specified annotation package.</li>
</ul>
</li>
</ol>
</div>
<div class="section level4">
<h4 id="parameter-guide-3">Parameter guide:<a class="anchor" aria-label="anchor" href="#parameter-guide-3"></a>
</h4>
<ul>
<li>
<code>inputPheno</code>: path to the merged phenotype–beta-value
<code>.RData</code> file (e.g. <code>phenoBetaT1.RData</code>).</li>
<li>
<code>outputLogs</code>: directory where model execution logs are
written.</li>
<li>
<code>outputRData</code>: directory where fitted GLM model objects
are saved.</li>
<li>
<code>outputPlots</code>: directory where plots (e.g. p-value
distributions, diagnostics) are written.</li>
<li>
<code>phenotypes</code>: comma-separated list of phenotype variables
to model independently.</li>
<li>
<code>covariates</code>: comma-separated list of covariates included
in each GLM.</li>
<li>
<code>factorVars</code>: subset of covariates treated as categorical
variables.</li>
<li>
<code>cpgPrefix</code>: prefix used to identify CpG probes (usually
<code>"cg"</code>).</li>
<li>
<code>cpgLimit</code>: maximum number of CpGs to analyse (useful for
testing); <code>NA</code> runs all CpGs.</li>
<li>
<code>nCores</code>: number of CPU cores used for parallel
processing.</li>
<li>
<code>plotWidth</code>, <code>plotHeight</code>,
<code>plotDPI</code>: dimensions and resolution of generated plots.</li>
<li>
<code>interactionTerm</code>: optional interaction term
(e.g. <code>phenotype:timepoint</code>); <code>NULL</code> disables
interactions.</li>
<li>
<code>libPath</code>: optional library path for HPC
environments.</li>
<li>
<code>glmLibs</code>: GLM backend library to use
(e.g. <code>glm2</code>).</li>
<li>
<code>prsMap</code>: optional PRS mapping file to include genetic
predictors.</li>
<li>
<code>summaryPval</code>: p-value threshold for generating summary
outputs; <code>NA</code> disables filtering.</li>
<li>
<code>summaryResidualSD</code>: logical flag to summarise residual
standard deviations.</li>
<li>
<code>saveSignificantCpGs</code>: whether to save CpGs passing
<code>significantCpGPval</code>.</li>
<li>
<code>significantCpGDir</code>: directory where significant CpG
tables are written.</li>
<li>
<code>significantCpGPval</code>: p-value threshold for defining
significant CpGs.</li>
<li>
<code>saveTxtSummaries</code>: whether to save per-model text
summaries.</li>
<li>
<code>chunkSize</code>: number of CpGs processed per chunk to
control memory usage.</li>
<li>
<code>summaryTxtDir</code>: directory for text-based model
summaries.</li>
<li>
<code>fdrThreshold</code>: FDR threshold used for adjusted
p-values.</li>
<li>
<code>padjmethod</code>: multiple-testing correction method
(e.g. <code>"fdr"</code>).</li>
<li>
<code>annotationPackage</code>: annotation package used to annotate
CpGs.</li>
<li>
<code>annotationCols</code>: comma-separated annotation fields to
extract.</li>
<li>
<code>annotatedGLMOut</code>: directory where annotated GLM result
tables are written.</li>
</ul>
</div>
<div class="section level4">
<h4 id="glm-structure-used-in-dnaepico-glm2">GLM structure used in dnaEPICO (<code>glm2</code>):<a class="anchor" aria-label="anchor" href="#glm-structure-used-in-dnaepico-glm2"></a>
</h4>
<p>For each CpG site ( cg_i ), dnaEPICO fits a generalised linear model
of the form:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="normal">Methylation</mtext><mo>*</mo><mrow><mi>c</mi><msub><mi>g</mi><mi>i</mi></msub></mrow><mo>=</mo><msub><mi>β</mi><mn>0</mn></msub><mo>+</mo><msub><mi>β</mi><mn>1</mn></msub><mo>⋅</mo><mtext mathvariant="normal">Phenotype</mtext><mo>+</mo><mo>∑</mo><mi>*</mi><msup><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msup><msub><mi>β</mi><mrow><mi>k</mi><mo>+</mo><mn>1</mn></mrow></msub><mo>⋅</mo><mtext mathvariant="normal">Covariate</mtext><mo>*</mo><mi>k</mi><mo>+</mo><mi>ε</mi></mrow><annotation encoding="application/x-tex">
\text{Methylation}*{cg_i} =
\beta_0 +
\beta_1 \cdot \text{Phenotype} +
\sum*{k=1}^{K} \beta_{k+1} \cdot \text{Covariate}*k +
\varepsilon
</annotation></semantics></math></p>
<p>where methylation is typically represented as
<strong>M-values</strong> (or Beta-values, depending on
configuration).</p>
</div>
<div class="section level4">
<h4 id="glm-without-prs-default">GLM without PRS (default):<a class="anchor" aria-label="anchor" href="#glm-without-prs-default"></a>
</h4>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtable><mtr><mtd columnalign="right" style="text-align: right"><msub><mtext mathvariant="normal">Methylation</mtext><mrow><mi>c</mi><msub><mi>g</mi><mi>i</mi></msub></mrow></msub><mo>=</mo><mspace width="0.278em"></mspace></mtd><mtd columnalign="left" style="text-align: left"><msub><mi>β</mi><mn>0</mn></msub><mo>+</mo><msub><mi>β</mi><mn>1</mn></msub><mo>⋅</mo><mtext mathvariant="normal">Phenotype</mtext><mo>+</mo><msub><mi>β</mi><mn>2</mn></msub><mo>⋅</mo><mtext mathvariant="normal">Age</mtext><mo>+</mo><msub><mi>β</mi><mn>3</mn></msub><mo>⋅</mo><mtext mathvariant="normal">Sex</mtext><mo>+</mo><msub><mi>β</mi><mn>4</mn></msub><mo>⋅</mo><mtext mathvariant="normal">Ethnicity</mtext><mo>+</mo></mtd></mtr><mtr><mtd columnalign="right" style="text-align: right"></mtd><mtd columnalign="left" style="text-align: left"><mi>+</mi><msub><mi>β</mi><mn>5</mn></msub><mo>⋅</mo><mtext mathvariant="normal">TraumaDefinition</mtext><mo>+</mo><msub><mi>β</mi><mn>6</mn></msub><mo>⋅</mo><mtext mathvariant="normal">Leukocytes</mtext><mo>+</mo><msub><mi>β</mi><mn>7</mn></msub><mo>⋅</mo><mtext mathvariant="normal">Epithelial.cells</mtext><mo>+</mo><mi>ε</mi></mtd></mtr></mtable><annotation encoding="application/x-tex">
\begin{aligned}
\text{Methylation}_{cg_i} = \;&amp;
\beta_0 +
\beta_1 \cdot \text{Phenotype} +
\beta_2 \cdot \text{Age} +
\beta_3 \cdot \text{Sex} +
\beta_4 \cdot \text{Ethnicity} + \\
&amp;+
\beta_5 \cdot \text{TraumaDefinition} +
\beta_6 \cdot \text{Leukocytes} +
\beta_7 \cdot \text{Epithelial.cells} +
\varepsilon
\end{aligned}
</annotation></semantics></math></p>
<p><strong>Mapping to dnaEPICO parameters</strong></p>
<ul>
<li>
<strong>Phenotype</strong>:<code>phenotypes</code>
</li>
<li>
<strong>Age, Sex, Ethnicity, TraumaDefinition, Leukocytes,
Epithelial.cells</strong>: <code>covariates</code>
</li>
<li>
<strong>Sex, Ethnicity, TraumaDefinition</strong>: converted to
<code>factorVars</code>
</li>
<li>(): residual error term</li>
</ul>
</div>
<div class="section level4">
<h4 id="glm-with-prs">GLM with PRS:<a class="anchor" aria-label="anchor" href="#glm-with-prs"></a>
</h4>
<p>When a PRS is supplied via <code>prsMap</code>, the model is
<strong>extended only for the matching phenotype</strong>.</p>
<p>From the example above:</p>
<pre><code><span><span class="va">prsMap</span> <span class="op">=</span> <span class="st">"Pheno1:PRS_1,Pheno2:PRS_2,Pheno3:PRS_3,Pheno4:PRS_4"</span></span></code></pre>
<p>This defines a <strong>one-to-one mapping</strong> between phenotype
and PRS. For example:</p>
<ul>
<li>Pheno1 with PRS_1</li>
</ul>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="normal">Methylation</mtext><mo>*</mo><mrow><mi>c</mi><msub><mi>g</mi><mi>i</mi></msub></mrow><mo>=</mo><msub><mi>β</mi><mn>0</mn></msub><mo>+</mo><msub><mi>β</mi><mn>1</mn></msub><mo>⋅</mo><mtext mathvariant="normal">Pheno1</mtext><mo>+</mo><mo>∑</mo><mi>*</mi><msup><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msup><msub><mi>β</mi><mrow><mi>k</mi><mo>+</mo><mn>1</mn></mrow></msub><mo>⋅</mo><mtext mathvariant="normal">Covariate</mtext><mo>*</mo><mi>k</mi><mo>+</mo><mi>β</mi><mo>*</mo><mtext mathvariant="normal">PRS</mtext><mo>⋅</mo><mtext mathvariant="normal">PRS_1</mtext><mo>+</mo><mi>ε</mi></mrow><annotation encoding="application/x-tex">
\text{Methylation}*{cg_i} =
\beta_0 +
\beta_1 \cdot \text{Pheno1} +
\sum*{k=1}^{K} \beta_{k+1} \cdot \text{Covariate}*k +
\beta*{\text{PRS}} \cdot \text{PRS_1} +
\varepsilon
</annotation></semantics></math></p>
<ul>
<li>Pheno2 with PRS_2</li>
</ul>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="normal">Methylation</mtext><mo>*</mo><mrow><mi>c</mi><msub><mi>g</mi><mi>i</mi></msub></mrow><mo>=</mo><msub><mi>β</mi><mn>0</mn></msub><mo>+</mo><msub><mi>β</mi><mn>1</mn></msub><mo>⋅</mo><mtext mathvariant="normal">Pheno2</mtext><mo>+</mo><mo>∑</mo><mi>*</mi><msup><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msup><msub><mi>β</mi><mrow><mi>k</mi><mo>+</mo><mn>1</mn></mrow></msub><mo>⋅</mo><mtext mathvariant="normal">Covariate</mtext><mo>*</mo><mi>k</mi><mo>+</mo><mi>β</mi><mo>*</mo><mtext mathvariant="normal">PRS</mtext><mo>⋅</mo><mtext mathvariant="normal">PRS_2</mtext><mo>+</mo><mi>ε</mi></mrow><annotation encoding="application/x-tex">
\text{Methylation}*{cg_i} =
\beta_0 +
\beta_1 \cdot \text{Pheno2} +
\sum*{k=1}^{K} \beta_{k+1} \cdot \text{Covariate}*k +
\beta*{\text{PRS}} \cdot \text{PRS_2} +
\varepsilon
</annotation></semantics></math></p>
</div>
<div class="section level4">
<h4 id="interaction-term">Interaction term<a class="anchor" aria-label="anchor" href="#interaction-term"></a>
</h4>
<p>If an interaction is specified (e.g. phenotype × timepoint):</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="normal">Methylation</mtext><mo>*</mo><mrow><mi>c</mi><msub><mi>g</mi><mi>i</mi></msub></mrow><mo>=</mo><mi>…</mi><mo>+</mo><mi>β</mi><mo>*</mo><mrow><mi>i</mi><mi>n</mi><mi>t</mi></mrow><mo>⋅</mo><mrow><mo stretchy="true" form="prefix">(</mo><mtext mathvariant="normal">Phenotype</mtext><mo>×</mo><mtext mathvariant="normal">Timepoint</mtext><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><mi>ε</mi></mrow><annotation encoding="application/x-tex">
\text{Methylation}*{cg_i} =
\ldots +
\beta*{int} \cdot (\text{Phenotype} \times \text{Timepoint}) +
\varepsilon
</annotation></semantics></math></p>
<p><strong>Mapping</strong></p>
<ul>
<li>Interaction definition: <code>interactionTerm</code>
</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="step-5-methylationglmm_t1t2">Step 5: <code>methylationGLMM_T1T2()</code><a class="anchor" aria-label="anchor" href="#step-5-methylationglmm_t1t2"></a>
</h3>
<p>This step is typically executed on <strong>HPC systems</strong> due
to memory and CPU requirements.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/methylationGLMM_T1T2.html">methylationGLMM_T1T2</a></span><span class="op">(</span></span>
<span>    inputPheno <span class="op">=</span> <span class="st">"rData/preprocessingPheno/mergeData/phenoBetaT1T2.RData"</span>,</span>
<span>    outputLogs <span class="op">=</span> <span class="st">"logs/"</span>,</span>
<span>    outputRData <span class="op">=</span> <span class="st">"rData/methylationGLMM_T1T2/models"</span>,</span>
<span>    outputPlots <span class="op">=</span> <span class="st">"figures/methylationGLMM_T1T2"</span>,</span>
<span>    personVar <span class="op">=</span> <span class="st">"person"</span>,</span>
<span>    timeVar <span class="op">=</span> <span class="st">"Timepoint"</span>,</span>
<span>    phenotypes <span class="op">=</span> <span class="st">"Pheno1,Pheno2,Pheno3,Pheno4"</span>,</span>
<span>    covariates <span class="op">=</span> <span class="st">"Sex,Age,Ethnicity,TraumaDefinition,Leukocytes,Epithelial.cells"</span>,</span>
<span>    factorVars <span class="op">=</span> <span class="st">"Sex,Ethnicity,TraumaDefinition,Timepoint"</span>,</span>
<span>    lmeLibs <span class="op">=</span> <span class="st">"lme4,lmerTest"</span>,</span>
<span>    prsMap <span class="op">=</span> <span class="st">"Pheno1:PRS_1,Pheno2:PRS_2,Pheno3:PRS_3,Pheno4:PRS_4"</span>,</span>
<span>    libPath <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    cpgPrefix <span class="op">=</span> <span class="st">"cg"</span>,</span>
<span>    cpgLimit <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>    nCores <span class="op">=</span> <span class="fl">32</span>,</span>
<span>    summaryPval <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>    plotWidth <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>    plotHeight <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>    plotDPI <span class="op">=</span> <span class="fl">150</span>,</span>
<span>    interactionTerm <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    saveSignificantInteractions <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    significantInteractionDir <span class="op">=</span> <span class="st">"preliminaryResults/cpgs/methylationGLMM_T1T2"</span>,</span>
<span>    significantInteractionPval <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    saveTxtSummaries <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    chunkSize <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>    summaryTxtDir <span class="op">=</span> <span class="st">"preliminaryResults/summary/methylationGLMM_T1T2/lmer"</span>,</span>
<span>    fdrThreshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>    padjmethod <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>    annotationPackage <span class="op">=</span> <span class="st">"IlluminaHumanMethylationEPICv2anno.20a1.hg38"</span>,</span>
<span>    annotationCols <span class="op">=</span> <span class="st">"Name,chr,pos,UCSC_RefGene_Group,UCSC_RefGene_Name,</span></span>
<span><span class="st">                      Relation_to_Island,GencodeV41_Group"</span>,</span>
<span>    annotatedLMEOut <span class="op">=</span> <span class="st">"data/methylationGLMM_T1T2"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="what-it-does-4">What it does:<a class="anchor" aria-label="anchor" href="#what-it-does-4"></a>
</h4>
<p>This wrapper executes the external script
<code>inst/scripts/methylationGLMM_T1T2.R</code> using a system call.
Conceptually, it:</p>
<ol style="list-style-type: decimal">
<li><p>Loads a merged <strong>longitudinal
phenotype–methylation</strong> object (<code>inputPheno</code>) created
by <code><a href="../reference/preprocessingPheno.html">preprocessingPheno()</a></code>.</p></li>
<li><p>Iterates over CpGs in chunks and fits a linear mixed-effects
model <a href="https://cran.r-project.org/web/packages/lme4/index.html" class="external-link">(lme4)</a>.</p></li>
<li><p>Fits <strong>one model per phenotype</strong>, accounting for
repeated measures using a subject-level random effect.</p></li>
<li><p>Models fixed effects for phenotype, time, covariates, and
optional interactions.</p></li>
<li><p>Optionally includes <strong>phenotype-specific PRS terms</strong>
based on <code>prsMap</code>.</p></li>
<li><p>Tests interaction terms (e.g. phenotype × timepoint).</p></li>
<li><p>Applies multiple-testing correction across CpGs.</p></li>
<li>
<p>Writes:</p>
<ul>
<li>fitted model objects to <code>outputRData</code>,</li>
<li>plots to <code>outputPlots</code>,</li>
<li>log files to <code>outputLogs</code>,</li>
<li>text summaries,</li>
<li>tables of significant interaction CpGs,</li>
<li>annotated result tables using the specified annotation package.</li>
</ul>
</li>
</ol>
</div>
<div class="section level4">
<h4 id="parameter-guide-4">Parameter guide:<a class="anchor" aria-label="anchor" href="#parameter-guide-4"></a>
</h4>
<ul>
<li>
<code>inputPheno</code>: path to the merged longitudinal
phenotype–methylation <code>.RData</code> file
(e.g. <code>phenoBetaT1T2.RData</code>).</li>
<li>
<code>outputLogs</code>: directory where execution logs are
written.</li>
<li>
<code>outputRData</code>: directory where fitted mixed-effects model
objects are saved.</li>
<li>
<code>outputPlots</code>: directory where model diagnostics and
summary plots are written.</li>
<li>
<code>personVar</code>: subject identifier used as a random effect
(e.g. participant ID).</li>
<li>
<code>timeVar</code>: variable defining the longitudinal time
dimension.</li>
<li>
<code>phenotypes</code>: comma-separated list of phenotype
variables.</li>
<li>
<code>covariates</code>: comma-separated list of covariates included
as fixed effects.</li>
<li>
<code>factorVars</code>: subset of variables treated as categorical
(typically includes <code>timeVar</code>).</li>
<li>
<code>lmeLibs</code>: mixed-effects modelling libraries used
(e.g. <code>lme4</code>, <code>lmerTest</code>).</li>
<li>
<code>prsMap</code>: optional phenotype-to-PRS mapping; PRS terms
are included <strong>only for mapped phenotypes</strong>.</li>
<li>
<code>libPath</code>: optional library path for HPC
environments.</li>
<li>
<code>cpgPrefix</code>: prefix used to identify CpG probes (usually
<code>"cg"</code>).</li>
<li>
<code>cpgLimit</code>: maximum number of CpGs to analyse;
<code>NA</code> analyses all CpGs.</li>
<li>
<code>nCores</code>: number of CPU cores used for parallel
processing.</li>
<li>
<code>summaryPval</code>: p-value threshold for generating
summaries; <code>NA</code> disables filtering.</li>
<li>
<code>plotWidth</code>, <code>plotHeight</code>,
<code>plotDPI</code>: dimensions and resolution of generated plots.</li>
<li>
<code>interactionTerm</code>: interaction term to test (commonly
<code>timeVar</code>); <code>NULL</code> disables interaction
testing.</li>
<li>
<code>saveSignificantInteractions</code>: logical flag indicating
whether significant interaction CpGs are saved.</li>
<li>
<code>significantInteractionDir</code>: directory where significant
interaction results are written.</li>
<li>
<code>significantInteractionPval</code>: p-value threshold defining
significant interactions.</li>
<li>
<code>saveTxtSummaries</code>: whether to save per-model text
summaries.</li>
<li>
<code>chunkSize</code>: number of CpGs processed per chunk to manage
memory usage.</li>
<li>
<code>summaryTxtDir</code>: directory for text-based model
summaries.</li>
<li>
<code>fdrThreshold</code>: FDR threshold applied to interaction
p-values.</li>
<li>
<code>padjmethod</code>: multiple-testing correction method
(e.g. <code>"fdr"</code>).</li>
<li>
<code>annotationPackage</code>: annotation package used to annotate
CpGs.</li>
<li>
<code>annotationCols</code>: comma-separated annotation fields to
extract.</li>
<li>
<code>annotatedLMEOut</code>: directory where annotated mixed-model
result tables are written.</li>
</ul>
</div>
<div class="section level4">
<h4 id="lmem-structure-used-in-dnaepico-lme4">LMEM structure used in dnaEPICO (<code>lme4</code>):<a class="anchor" aria-label="anchor" href="#lmem-structure-used-in-dnaepico-lme4"></a>
</h4>
<p>For longitudinal analyses, the phenotype-specific structure remains,
with additional random and interaction terms:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtable><mtr><mtd columnalign="right" style="text-align: right"><mtext mathvariant="normal">Methylation</mtext><mo>*</mo><mrow><mi>c</mi><msub><mi>g</mi><mi>i</mi></msub></mrow><mo>=</mo><mspace width="0.278em"></mspace></mtd><mtd columnalign="left" style="text-align: left"><msubsup><mi>β</mi><mn>0</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>p</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>+</mo><msubsup><mi>β</mi><mn>1</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>p</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>⋅</mo><mi>p</mi><mo>+</mo><msubsup><mi>β</mi><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>p</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>⋅</mo><mtext mathvariant="normal">Timepoint</mtext><mo>+</mo><msubsup><mi>β</mi><mn>3</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>p</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>⋅</mo><mrow><mo stretchy="true" form="prefix">(</mo><mi>p</mi><mo>×</mo><mtext mathvariant="normal">Timepoint</mtext><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo></mtd></mtr><mtr><mtd columnalign="right" style="text-align: right"></mtd><mtd columnalign="left" style="text-align: left"><mi>+</mi><mi>β</mi><mo>*</mo><msup><mtext mathvariant="normal">PRS</mtext><mrow><mo stretchy="true" form="prefix">(</mo><mi>p</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>⋅</mo><mtext mathvariant="normal">PRS</mtext><mo>*</mo><mi>p</mi><mo>+</mo><mi>b</mi><mo>*</mo><mtext mathvariant="normal">person</mtext><mo>+</mo><msup><mi>ε</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>p</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup></mtd></mtr></mtable><annotation encoding="application/x-tex">
\begin{aligned}
\text{Methylation}*{cg_i} = \;&amp;
\beta_0^{(p)} +
\beta_1^{(p)} \cdot p +
\beta_2^{(p)} \cdot \text{Timepoint} +
\beta_3^{(p)} \cdot (p \times \text{Timepoint}) + \\
&amp;+
\beta*{\text{PRS}}^{(p)} \cdot \text{PRS}*p +
b*{\text{person}} +
\varepsilon^{(p)}
\end{aligned}
</annotation></semantics></math></p>
<p>where (_p) is included <strong>only if it is defined for phenotype
(p)</strong>. The logic follows that of
<code><a href="../reference/methylationGLM_T1.html">methylationGLM_T1()</a></code>.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="makefile-pipeline-concepts">Makefile pipeline concepts<a class="anchor" aria-label="anchor" href="#makefile-pipeline-concepts"></a>
</h2>
<ul>
<li>
<code>Makefile.model.pipeline</code>: user configuration (models,
variables, paths)</li>
<li>
<code>Makefile.rules.pipeline</code>: pipeline targets and
rules</li>
</ul>
<div class="section level3">
<h3 id="extracting-makefile-model-pipeline-into-a-working-directory">Extracting <code>Makefile.model.pipeline</code> into a working
directory<a class="anchor" aria-label="anchor" href="#extracting-makefile-model-pipeline-into-a-working-directory"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Copies the example Makefile into the current directory</span></span>
<span><span class="fu"><a href="../reference/extractMake.html">extractMake</a></span><span class="op">(</span>destDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>, overwrite <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>The internal template <code>Makefile.model.pipeline</code> is copied
and renamed to <code>Makefile</code> in the destination directory.</p>
</div>
<div class="section level3">
<h3 id="make-targets-f3-f4-f3lme-and-all">Make targets: f3, f4, f3lme and all<a class="anchor" aria-label="anchor" href="#make-targets-f3-f4-f3lme-and-all"></a>
</h3>
<p>Common grouped targets:</p>
<ul>
<li>
<code>f3</code>: Steps 1–3 (preprocessing + SVA + merge)</li>
<li>
<code>f4</code>: Steps 1–4 (adds GLM)</li>
<li>
<code>f3lme</code>: Steps 1-3 and 5 (adds LME4)</li>
<li>
<code>all</code>: Steps 1–5</li>
</ul>
<div class="section level4">
<h4 id="commands">Commands<a class="anchor" aria-label="anchor" href="#commands"></a>
</h4>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="do">## Show status of pipeline outputs</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>make status MODEL<span class="ot">=</span>model1</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="do">## Run steps 1–3 for one model</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>make f3 MODEL<span class="ot">=</span>model1</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a><span class="do">## Run steps 1–4 for one model</span></span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>make f4 MODEL<span class="ot">=</span>model1</span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a><span class="do">## Run steps 1–3 and 5 for one model</span></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a>make f3lme MODEL<span class="ot">=</span>model1</span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a></span>
<span id="cb15-13"><a href="#cb15-13" tabindex="-1"></a><span class="do">## Run full pipeline</span></span>
<span id="cb15-14"><a href="#cb15-14" tabindex="-1"></a>make all MODEL<span class="ot">=</span>model1</span>
<span id="cb15-15"><a href="#cb15-15" tabindex="-1"></a></span>
<span id="cb15-16"><a href="#cb15-16" tabindex="-1"></a><span class="do">## Run multiple models</span></span>
<span id="cb15-17"><a href="#cb15-17" tabindex="-1"></a>make models <span class="co"># run the entire pipeline for all models in parallel.</span></span>
<span id="cb15-18"><a href="#cb15-18" tabindex="-1"></a>make f3_models <span class="co"># run Steps 1–3 in parallel for all models.</span></span>
<span id="cb15-19"><a href="#cb15-19" tabindex="-1"></a>make f4_models <span class="co"># run Steps 1–4 in parallel for all models.</span></span>
<span id="cb15-20"><a href="#cb15-20" tabindex="-1"></a>make f3lme_models <span class="co"># run Steps 1–3 + LME in parallel for all models</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="makefile-modelling-steps-hpc">Makefile Modelling steps (HPC)<a class="anchor" aria-label="anchor" href="#makefile-modelling-steps-hpc"></a>
</h3>
<p>This Makefile defines <strong>how dnaEPICO models are executed at
scale</strong>, controlling:</p>
<ul>
<li>which phenotype files are used,</li>
<li>which statistical models are run,</li>
<li>how GLM (<code>glm2</code>) and GLMM (<code>lme4</code>) are
parameterised,</li>
<li>how results are organised and reproduced across models.</li>
</ul>
<p>Rather than hard-coding parameters in R scripts, dnaEPICO uses a
<strong>declarative Makefile layer</strong> so the same pipeline can be
re-run with different configurations.</p>
<pre class="make"><code># ===============================================
# USER CONFIGURATION
# ===============================================
MAKEFLAGS += --output-sync

# ===============================================
# Run all models 
# ===============================================
models:
    @echo "Running models in parallel: $(MODELS)"
    @$(MAKE) -j $(words $(MODELS)) $(addprefix run-, $(MODELS))

$(foreach m,$(MODELS),$(eval run-$(m):; \
    @echo "&gt;&gt;&gt; Starting full pipeline for: $(m)"; \
    $(MAKE) -j1 all MODEL=$(m); \
    @echo "&lt;&lt;&lt; Finished full pipeline for: $(m)" \
))

default: all

all: \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/m_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/beta_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/cn_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(DATA_DIR)/$(MODEL)/$(STEP2)/summary_full_sva2.txt \
    $(DATA_DIR)/$(MODEL)/$(STEP1)/phenoLC.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT2.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1T2.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP4)/annotatedGLM.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP5)/annotatedLME.csv \
    $(STEP6)/$(MODEL)/DNAm.pdf
    
# ===============================================
# Group target: first3 (Steps 1 to 3)
# ===============================================
.PHONY: f3
FIRST3 = \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/m_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/beta_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/cn_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(DATA_DIR)/$(MODEL)/$(STEP2)/summary_full_sva2.txt \
    $(DATA_DIR)/$(MODEL)/$(STEP1)/phenoLC.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT2.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1T2.csv

f3: $(FIRST3) $(STEP6)/$(MODEL)/DNAm.pdf

# ===============================================
# Run f3 (Steps 1 to 3) for all models in parallel
# ===============================================
f3_models:
    @echo "Running f3 (Steps 1–3) in parallel: $(MODELS)"
    @$(MAKE) -j $(words $(MODELS)) $(addprefix runf3-, $(MODELS))

$(foreach m,$(MODELS),$(eval runf3-$(m):;  \
    @echo "&gt;&gt;&gt; Starting f3 for: $(m)"; \
    $(MAKE) -j1 f3 MODEL=$(m); \
    @echo "&lt;&lt;&lt; Finished f3 for: $(m)" \
))

# ===============================================
# Group target: f4 (Steps 1 to 4)
# ===============================================
.PHONY: f4
FIRST4 = \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/m_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/beta_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/cn_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(DATA_DIR)/$(MODEL)/$(STEP2)/summary_full_sva2.txt \
    $(DATA_DIR)/$(MODEL)/$(STEP1)/phenoLC.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT2.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1T2.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP4)/annotatedGLM.csv

f4: $(FIRST4) $(STEP6)/$(MODEL)/DNAm.pdf

# ===============================================
# Run f4 (Steps 1 to 4 only) for all models in parallel
# ===============================================
f4_models:
    @echo "Running f4 (Steps 1–4) in parallel: $(MODELS)"
    @$(MAKE) -j $(words $(MODELS)) $(addprefix runf4-, $(MODELS))

$(foreach m,$(MODELS),$(eval runf4-$(m):; \
    @echo "&gt;&gt;&gt; Starting f4 for: $(m)"; \
    $(MAKE) -j1 f4 MODEL=$(m); \
    @echo "&lt;&lt;&lt; Finished f4 for: $(m)" \
))

# ===============================================
# Group target: f3lme (Steps 1 to 3 + LME)
# ===============================================
.PHONY: f3lme
F3LME = \
  $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/m_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/beta_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/cn_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
    $(DATA_DIR)/$(MODEL)/$(STEP2)/summary_full_sva2.txt \
    $(DATA_DIR)/$(MODEL)/$(STEP1)/phenoLC.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT2.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1T2.csv \
    $(DATA_DIR)/$(MODEL)/$(STEP5)/annotatedLME.csv

f3lme: $(F3LME) $(STEP6)/$(MODEL)/DNAm.pdf

# ===============================================
# Run f3lme (Steps 1 to 3 + LME) for all models in parallel
# ===============================================

f3lme_models:
    @echo "Running f3lme (Steps 1–3 + LME) in parallel: $(MODELS)"
    @$(MAKE) -j $(words $(MODELS)) $(addprefix runf3lme-, $(MODELS))

$(foreach m,$(MODELS),$(eval runf3lme-$(m):; \
    @echo "&gt;&gt;&gt; Starting f3lme for: $(m)"; \
    $(MAKE) -j1 f3lme MODEL=$(m); \
    @echo "&lt;&lt;&lt; Finished f3lme for: $(m)" \
))

# =================================================================================================================================
# Step 1: Minfi Preprocessing
# =================================================================================================================================
step1:
    @echo "Running Step 1 for $(MODEL)"
    $(MAKE) preprocessingMinfiEwasWater MODEL=$(MODEL)

$(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/m_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
$(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/beta_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
$(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/cn_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData \
$(DATA_DIR)/$(MODEL)/$(STEP1)/phenoLC.csv:
    Rscript -e "library(dnapipeR); preprocessingMinfiEwasWater( \
      phenoFile = '$(PHENO_FILE)', \
      sepType = '$(SEP_TYPE)', \
      idatFolder = '$(DATA_DIR)/$(STEP1)/$(IDAT_DIR)/', \
      outputLogs = '$(LOGS_DIR)/$(MODEL)', \
      nSamples = $(N_SAMPLES), \
      SampleID = '$(SAMPLE_ID)', \
      arrayType = '$(ARRAY_TYPE)', \
      annotationVersion = '$(ANNOTATION_VERSION)', \
      scriptLabel = '$(MODEL)/$(STEP1)', \
      baseDataFolder = '$(RDATA_DIR)', \
      tiffWidth = $(TIFF_WIDTH), \
      tiffHeight = $(TIFF_HEIGHT), \
      tiffRes = $(TIFF_RES), \
      qcCutoff = $(QC_CUTOFF), \
      detPtype = '$(DET_PTYPE)', \
      detPThreshold = $(DET_PTHRESHOLD), \
      funnormSeed = $(FUNNORM_SEED), \
      sexColumn = '$(SEX_COLUMN)', \
      pvalThreshold = $(PVAL_THRESHOLD), \
      chrToRemove = '$(CHR_TO_REMOVE)', \
      snpsToRemove = '$(SNPS_TO_REMOVE)', \
      crossReactivePath = '$(CROSS_REACTIVE_FILE)', \
      mafThreshold = $(MAF_THRESHOLD), \
      plotGroupVar = '$(PLOT_GROUP_VAR)', \
      lcRef = '$(LC_REF)', \
      phenoOrder = '$(PHENO_ORDER)', \
      lcPhenoDir = '$(DATA_DIR)/$(MODEL)/$(STEP1)' )"

# =================================================================================================================================
# Step 2: Surrogate Variable Analysis
# =================================================================================================================================
$(DATA_DIR)/$(MODEL)/$(STEP2)/summary_full_sva2.txt:
    Rscript -e "library(dnapipeR); svaEnmix( \
      phenoFile = '$(DATA_DIR)/$(MODEL)/$(STEP1)/phenoLC.csv', \
      rgsetData = '$(RDATA_DIR)/$(MODEL)/$(STEP1)/$(OBJ_DIR)/RGSet.RData', \
      sepType = '$(SEP_TYPE)', \
      outputLogs = '$(LOGS_DIR)/$(MODEL)', \
      nSamples = $(N_SAMPLES), \
      SampleID = '$(SAMPLE_ID)', \
      arrayType = '$(ARRAY_TYPE)', \
      annotationVersion = '$(ANNOTATION_VERSION)', \
      SentrixIDColumn = '$(SENTRIX_ID_COLUMN)', \
      SentrixPositionColumn = '$(SENTRIX_POSITION_COLUMN)', \
      ctrlSvaPercVar = $(CTRL_SVA_PERC_VAR), \
      ctrlSvaFlag = $(CTRL_SVA_FLAG), \
      scriptLabel = '$(MODEL)/$(STEP2)', \
      tiffWidth = $(TIFF_WIDTH), \
      tiffHeight = $(TIFF_HEIGHT), \
      tiffRes = $(TIFF_RES) )"

# =================================================================================================================================
# Step 3: Merge Phenotype
# =================================================================================================================================
$(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1.csv \
$(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT2.csv \
$(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1T2.csv: 
    Rscript -e "library(dnapipeR); preprocessingPheno( \
      phenoFile = '$(DATA_DIR)/$(MODEL)/$(STEP1)/phenoLC.csv', \
      sepType = '$(SEP_TYPE)', \
      betaPath = '$(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/beta_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData', \
      mPath = '$(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/m_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData', \
      cnPath = '$(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/cn_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData', \
      SampleID = '$(SAMPLE_ID)', \
      timeVar = '$(TIME_VAR)', \
      timepoints = '$(TIMEPOINTS)', \
      combineTimepoints = '$(COMBINE_TIMEPOINTS)', \
      outputPheno = '$(DATA_DIR)/$(MODEL)/$(STEP3)', \
      outputRData = '$(RDATA_DIR)/$(MODEL)/$(STEP3)/$(METRICS_DIR)', \
      outputRDataMerge = '$(RDATA_DIR)/$(MODEL)/$(STEP3)/$(MERGE_DIR)', \
      outputLogs = '$(LOGS_DIR)/$(MODEL)', \
      sexColumn = '$(SEX_COLUMN)', \
      outputDir = '$(DATA_DIR)/$(MODEL)/$(STEP3)' )"  
      
# =================================================================================================================================
# Step 4: GLM for T1
# =================================================================================================================================
$(DATA_DIR)/$(MODEL)/$(STEP4)/annotatedGLM.csv:
    Rscript -e "library(dnapipeR); methylationGLM_T1( \
      inputPheno = '$(RDATA_DIR)/$(MODEL)/$(STEP3)/$(MERGE_DIR)/phenoBetaT1.RData', \
      outputLogs = '$(LOGS_DIR)/$(MODEL)', \
      outputRData = '$(RDATA_DIR)/$(MODEL)/$(STEP4)/$(MODEL_DIR)', \
      outputPlots = '$(FIGURES_DIR)/$(MODEL)/$(STEP4)/', \
      phenotypes = '$(PHENOTYPES)', \
      covariates = '$(COVARIATES)', \
      factorVars = '$(FACTOR_VARS)', \
      cpgPrefix = '$(CPG_PREFIX)', \
      cpgLimit = $(CPG_LIMIT), \
      nCores = $(N_CORES), \
      plotWidth = $(TIFF_WIDTH), \
      plotHeight = $(TIFF_HEIGHT), \
      plotDPI = $(TIFF_RES), \
      interactionTerm = $(INTERACTION_GLM_ARG), \
      libPath = $(R_DIR_ARG), \
      glmLibs = '$(GLM_LIBS)', \
      prsMap = $(PRS_MAP_ARG), \
      summaryPval = $(SUMMARY_PVAL), \
      summaryResidualSD = $(SUMMARY_RESIDUAL_SD), \
      saveSignificantCpGs = $(SAVE_SIGNIFICANT_CPGS), \
      significantCpGDir = '$(RESULTS_DIR)/$(MODEL)/$(CPG_DIR)/$(STEP4)', \
      significantCpGPval = $(SIGNIFICANT_CPG_PVAL), \
      saveTxtSummaries = $(SAVE_TXT_SUMMARIES), \
      chunkSize = $(CHUNK_SIZE), \
      summaryTxtDir = '$(RESULTS_DIR)/$(MODEL)/$(SUMMARY_DIR)/$(STEP4)', \
      fdrThreshold = $(FDR_THRESHOLD), \
      padjmethod = '$(PADJ_METHOD)', \
      annotationPackage = '$(ANNOTATION_PACKAGE)', \
      annotationCols = '$(ANNOTATION_COLS)', \
      annotatedGLMOut = '$(DATA_DIR)/$(MODEL)/$(STEP4)' ) "

# =================================================================================================================================
# Step 5: LME for T1 vs T2 (Longitudinal Analysis)
# =================================================================================================================================

PERSON_VAR = person
LME_LIBS = lme4,lmerTest
SAVE_SIGNIFICANT_INTERACTIONS = TRUE
SIGNIFICANT_INTERACTION_PVAL = 0.00001


$(DATA_DIR)/$(MODEL)/$(STEP5)/annotatedLME.csv:
    Rscript -e "library(dnapipeR); methylationGLMM_T1T2( \
      inputPheno = '$(RDATA_DIR)/$(MODEL)/$(STEP3)/$(MERGE_DIR)/phenoBetaT1T2.RData', \
      outputLogs = '$(LOGS_DIR)/$(MODEL)', \
      outputRData = '$(RDATA_DIR)/$(MODEL)/$(STEP5)/$(MODEL_DIR)', \
      outputPlots = '$(FIGURES_DIR)/$(MODEL)/$(STEP5)', \
      personVar = '$(PERSON_VAR)', \
      timeVar = '$(TIME_VAR)', \
      phenotypes = '$(PHENOTYPES)', \
      covariates = '$(COVARIATES)', \
      factorVars = '$(FACTOR_VARS)', \
      lmeLibs = '$(LME_LIBS)', \
      prsMap = $(PRS_MAP_ARG), \
      libPath = $(R_DIR_ARG), \
      cpgPrefix = '$(CPG_PREFIX)', \
      cpgLimit = $(CPG_LIMIT), \
      nCores = $(N_CORES), \
      summaryPval = $(SUMMARY_PVAL), \
      plotWidth = $(TIFF_WIDTH), \
      plotHeight = $(TIFF_HEIGHT), \
      plotDPI = $(TIFF_RES), \
      interactionTerm = $(INTERACTION_LME_ARG), \
      saveSignificantInteractions = $(SAVE_SIGNIFICANT_INTERACTIONS), \
      significantInteractionDir = '$(RESULTS_DIR)/$(MODEL)/$(CPG_DIR)/$(STEP5)', \
      significantInteractionPval = $(SIGNIFICANT_INTERACTION_PVAL), \
      saveTxtSummaries = $(SAVE_TXT_SUMMARIES), \
      chunkSize = $(CHUNK_SIZE), \
      summaryTxtDir = '$(RESULTS_DIR)/$(MODEL)/$(SUMMARY_DIR)/$(STEP5)', \
      fdrThreshold = $(FDR_THRESHOLD), \
      padjmethod = '$(PADJ_METHOD)', \
      annotationPackage = '$(ANNOTATION_PACKAGE)', \
      annotationCols = '$(ANNOTATION_COLS)', \
      annotatedLMEOut = '$(DATA_DIR)/$(MODEL)/$(STEP5)' )"

# =================================================================================================================================
# Step 6: Final Report
# =================================================================================================================================
REPORT_INPUTS = \
    $(RDATA_DIR)/$(MODEL)/$(STEP3)/$(MERGE_DIR)/phenoBetaT1.RData \
    $(DATA_DIR)/$(MODEL)/$(STEP1)/phenoLC.csv

$(STEP6)/$(MODEL)/DNAm.pdf: $(REPORT_INPUTS)
    mkdir -p $(STEP6)/$(MODEL) $(LOGS_DIR)/$(MODEL)
    Rscript -e "library(dnapipeR); dnamReport( \
      output = 'DNAm.pdf', \
      outputDir = '$(STEP6)/$(MODEL)', \
      qcDir = '$(FIGURES_DIR)/$(MODEL)/$(STEP1)/$(ENMIX_DIR)', \
      preprocessingDir = '$(FIGURES_DIR)/$(MODEL)/$(STEP1)/$(QC_DIR)', \
      postprocessingDir = '$(FIGURES_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)', \
      svaDir = '$(FIGURES_DIR)/$(MODEL)/$(STEP2)/$(SVA_DIR)', \
      glmDir = '$(FIGURES_DIR)/$(MODEL)/$(STEP4)', \
      glmmDir = '$(FIGURES_DIR)/$(MODEL)/$(STEP5)', \
      reportTitle = 'DNA methylation – $(MODEL)', \
      author = 'School of Biomedical Sciences' )" \
      &gt; $(LOGS_DIR)/$(MODEL)/report.log 2&gt;&amp;1
    @echo "Report built: $(STEP6)/$(MODEL)/DNAm.pdf"

# =================================================================================================================================
# Clean up outputs
# =================================================================================================================================
clean:
ifeq ($(MODEL),all)
    @echo "===== Cleaning all models ====="
    @for m in $(MODELS); do \
      echo "Cleaning model $$m ..."; \
      rm -rf \
        $(DATA_DIR)/$$m \
        $(RDATA_DIR)/$$m \
        $(STEP6)/$$m \
        $(LOGS_DIR)/$$m \
        $(FIGURES_DIR)/$$m \
        $(RESULTS_DIR)/$$m; \
    done
    @echo "Clean completed for all models. Kept $(DATA_DIR)/$(STEP1)/."
else
    @echo "Cleaning outputs for: $(MODEL)"
    rm -rf \
        $(DATA_DIR)/$(MODEL) \
        $(RDATA_DIR)/$(MODEL) \
        $(STEP6)/$(MODEL) \
        $(LOGS_DIR)/$(MODEL) \
        $(FIGURES_DIR)/$(MODEL) \
        $(RESULTS_DIR)/$(MODEL)
    @echo "Clean completed for: $(MODEL)"
endif

# =================================================================================================================================
# Status Check Target
# =================================================================================================================================
status:
    @echo "===== Pipeline Status ====="
    @test -e $(RDATA_DIR)/$(MODEL)/$(STEP1)/$(METRICS_DIR)/beta_NomFilt_MSetF_Flt_Rxy_Ds_Rc.RData &amp;&amp; echo "? Step 1: preprocessingMinfiEwasWater done" || echo "? Step 1: preprocessingMinfiEwasWater outcome file missing"
    @test -e $(DATA_DIR)/$(MODEL)/$(STEP2)/summary_full_sva2.txt &amp;&amp; echo "? Step 2: SVA done" || echo "? Step 2: SVA outcome file missing"
    @test -e $(DATA_DIR)/$(MODEL)/$(STEP3)/phenoT1T2.csv &amp;&amp; echo "? Step 3: preprocessingPheno done" || echo "? Step 3: preprocessingPheno outcome file missing"
    @test -e $(DATA_DIR)/$(MODEL)/$(STEP4)/annotatedGLM.csv &amp;&amp; echo "? Step 4: methylationGLM_T1 done" || echo "? Step 4: methylationGLM_T1 outcome file missing"
    @test -e $(DATA_DIR)/$(MODEL)/$(STEP5)/annotatedLME.csv &amp;&amp; echo "? Step 5: methylationGLMM_T1T2 done" || echo "? Step 5: methylationGLMM_T1T2 outcome file missing"
    @test -e $(STEP6)/$(MODEL)/DNAm.pdf &amp;&amp; echo "? Step 6: Report generated" || echo "? Step 6: Report missing"
    @echo "============================"

.PHONY: all clean status models f3 f4 f3lme f3_models f4_models f3lme_models
</code></pre>
<div class="section level4">
<h4 id="model-selection">Model selection<a class="anchor" aria-label="anchor" href="#model-selection"></a>
</h4>
<pre class="make"><code>MODEL ?= model1
MODELS = modelA modelB modelC</code></pre>
<ul>
<li>
<code>MODEL</code> defines <strong>a single pipeline
configuration</strong>
</li>
<li>
<code>MODELS</code> defines <strong>multiple independent
configurations</strong>
</li>
</ul>
<p>The same pipeline can therefore be run in two modes:</p>
<p><strong>Single-model execution</strong></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">make</span> f4 MODEL=modelA</span></code></pre></div>
<p><strong>Parallel multi-model execution</strong></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">make</span> f4_models</span></code></pre></div>
<p>In parallel mode, <strong>each model runs as a fully independent
pipeline</strong>, with its own inputs, logs, figures, and outputs. No
files are shared between models except the raw inputs (IDATs), unless
explicitly specified.</p>
</div>
</div>
<div class="section level3">
<h3 id="per-model-overrides">Per-model overrides<a class="anchor" aria-label="anchor" href="#per-model-overrides"></a>
</h3>
<pre class="make"><code>ifeq ($(MODEL), modelA)
  PHENO_FILE = data/preprocessingMinfiEwasWater/phenoA.csv
...
else
  PHENO_FILE = data/preprocessingMinfiEwasWater/pheno.csv
endif</code></pre>
<p>This block <strong>dynamically replaces variables</strong> based on
the selected model.</p>
<ul>
<li>The Makefile is evaluated <strong>at runtime</strong>
</li>
<li>When <code>MODEL=modelA</code>, <code>PHENO_FILE</code> is replaced
before any step runs</li>
<li>The rest of the pipeline remains unchanged</li>
</ul>
<p>This allows users to:</p>
<ul>
<li>compare cohorts or subgroup definitions,</li>
<li>test alternative phenotype encodings,</li>
<li>run sensitivity analyses,</li>
</ul>
<p>without duplicating code or Makefiles.</p>
<div class="section level4">
<h4 id="directory-and-step-conventions">Directory and step conventions<a class="anchor" aria-label="anchor" href="#directory-and-step-conventions"></a>
</h4>
<pre class="make"><code><span><span class="va">STEP1</span> <span class="op">=</span> <span class="va">preprocessingMinfiEwasWater</span></span>
<span><span class="va">STEP2</span> <span class="op">=</span> <span class="va">svaEnmix</span></span>
<span><span class="va">STEP3</span> <span class="op">=</span> <span class="va">preprocessingPheno</span></span>
<span><span class="va">STEP4</span> <span class="op">=</span> <span class="va">methylationGLM_T1</span></span>
<span><span class="va">STEP5</span> <span class="op">=</span> <span class="va">methylationGLMM_T1T2</span></span></code></pre>
<p>Each step writes outputs into <strong>step-specific
directories</strong>. When running in parallel, directory paths are
automatically namespaced by model, ensuring isolation between runs.</p>
<p>This design allows:</p>
<ul>
<li>restarting individual steps,</li>
<li>debugging failed models independently,</li>
<li>mixing local and HPC execution safely.</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="global-parameters">Global parameters<a class="anchor" aria-label="anchor" href="#global-parameters"></a>
</h3>
<pre class="make"><code><span><span class="va">PHENO_FILE</span> <span class="op">=</span> <span class="va">...</span></span>
<span><span class="va">SAMPLE_ID</span> <span class="op">=</span> <span class="va">...</span></span>
<span><span class="va">TIME_VAR</span> <span class="op">=</span> <span class="va">...</span></span></code></pre>
<p>These parameters are <strong>shared across preprocessing and
modelling</strong>.</p>
<p>Their definitions and expected formats are explained in the <em>Local
use</em> section and are not repeated here.</p>
<div class="section level4">
<h4 id="specific-parameters">Specific parameters<a class="anchor" aria-label="anchor" href="#specific-parameters"></a>
</h4>
<pre class="make"><code>PHENOTYPES = TreatmentGroup
COVARIATES = Sex
FACTOR_VARS = Sex,TreatmentGroup
N_CORES = 64
CHUNK_SIZE = 10000</code></pre>
<p>These parameters control <strong>how each model is fit</strong>. In
parallel execution:</p>
<ul>
<li>
<code>N_CORES</code> controls <strong>intra-model
parallelism</strong> (within R)</li>
<li>
<code>MODELS</code> controls <strong>inter-model
parallelism</strong> (within Make)</li>
</ul>
<p>Their definitions and expected formats are explained in the <em>Local
use</em> section and are not repeated here.</p>
</div>
<div class="section level4">
<h4 id="argument-normalisation">Argument normalisation<a class="anchor" aria-label="anchor" href="#argument-normalisation"></a>
</h4>
<pre class="make"><code>ifeq ($(PRS_MAP),NULL)
  PRS_MAP_ARG = NULL
else
  PRS_MAP_ARG = '$(PRS_MAP)'
endif
if ...
</code></pre>
<p>This section ensures that optional arguments are passed correctly to
R scripts. It prevents accidental interpretation of the string
<code>"NULL"</code> as a real value and guarantees consistent behaviour
across local and cluster environments.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="reporting">Reporting<a class="anchor" aria-label="anchor" href="#reporting"></a>
</h2>
<p><code><a href="../reference/dnamReport.html">dnamReport()</a></code> generates a PDF report by rendering
<code>DNAm.Rmd</code> through a runner script (<code>DNAm.R</code>)
shipped in <code>inst/scripts</code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/dnamReport.html">dnamReport</a></span><span class="op">(</span></span>
<span>  output <span class="op">=</span> <span class="st">"DNAm_Report.pdf"</span>,</span>
<span>  outputDir <span class="op">=</span> <span class="st">"reports"</span>,</span>
<span>  qcDir <span class="op">=</span> <span class="st">"figures/preprocessingMinfiEwasWater/enMix"</span>,</span>
<span>  preprocessingDir <span class="op">=</span> <span class="st">"figures/preprocessingMinfiEwasWater/qc"</span>,</span>
<span>  postprocessingDir <span class="op">=</span> <span class="st">"figures/preprocessingMinfiEwasWater/metrics"</span>,</span>
<span>  svaDir <span class="op">=</span> <span class="st">"figures/svaEnmix/sva"</span>,</span>
<span>  glmDir <span class="op">=</span> <span class="st">"figures/methylationGLM_T1"</span>,</span>
<span>  glmmDir <span class="op">=</span> <span class="st">"figures/methylationGLMM_T1T2"</span>,</span>
<span>  reportTitle <span class="op">=</span> <span class="st">"DNA methylation"</span>,</span>
<span>  author <span class="op">=</span> <span class="st">"School of Biomedical Sciences"</span>,</span>
<span>  date <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/format.html" class="external-link">format</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html" class="external-link">Sys.Date</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"%B %d, %Y"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="troubleshooting">Troubleshooting<a class="anchor" aria-label="anchor" href="#troubleshooting"></a>
</h2>
<ul>
<li>If a step “fails”, check logs in <code>logs/</code>.</li>
<li>If Make targets are missing, confirm:
<ul>
<li>correct <code>include Makefile.rules.pipeline</code>
</li>
</ul>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2026-01-23 12:28:41 AEST"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 1.753 secs</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; R version 4.4.1 (2024-06-14 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64</span></span>
<span><span class="co">#&gt; Running under: Windows 11 x64 (build 26100)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_Australia.utf8  LC_CTYPE=English_Australia.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_Australia.utf8 LC_NUMERIC=C                      </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_Australia.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Australia/Brisbane</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] BiocStyle_2.34.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] cli_3.6.5           knitr_1.51          rlang_1.1.6        </span></span>
<span><span class="co">#&gt;  [4] xfun_0.52           otel_0.2.0          textshaping_1.0.4  </span></span>
<span><span class="co">#&gt;  [7] jsonlite_2.0.0      htmltools_0.5.8.1   ragg_1.5.0         </span></span>
<span><span class="co">#&gt; [10] sass_0.4.10         rmarkdown_2.30      evaluate_1.0.5     </span></span>
<span><span class="co">#&gt; [13] jquerylib_0.1.4     fastmap_1.2.0       yaml_2.3.10        </span></span>
<span><span class="co">#&gt; [16] lifecycle_1.0.5     bookdown_0.46       BiocManager_1.30.27</span></span>
<span><span class="co">#&gt; [19] compiler_4.4.1      fs_1.6.6            htmlwidgets_1.6.4  </span></span>
<span><span class="co">#&gt; [22] rstudioapi_0.18.0   systemfonts_1.3.1   digest_0.6.37      </span></span>
<span><span class="co">#&gt; [25] R6_2.6.1            bslib_0.9.0         tools_4.4.1        </span></span>
<span><span class="co">#&gt; [28] pkgdown_2.2.0       cachem_1.1.0        desc_1.4.3</span></span></code></pre>
<div class="section level3">
<h3 id="acknowledgement">Acknowledgement<a class="anchor" aria-label="anchor" href="#acknowledgement"></a>
</h3>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.20/biocthis" class="external-link">biocthis</a></em>.</p>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>,
with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> and
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
running behind the scenes.</p>
</div>
<div class="section level3">
<h3 id="required-knowledge">Required knowledge<a class="anchor" aria-label="anchor" href="#required-knowledge"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.20/dnaEPICO" class="external-link">dnaEPICO</a></em>
is built on core Bioconductor infrastructure for high-dimensional
genomic data, with a focus on Illumina EPICv2 DNA methylation
arrays.</p>
<p>Preprocessing and quality control are performed using established
Bioconductor tools, including <em><a href="https://bioconductor.org/packages/3.20/minfi" class="external-link">minfi</a></em>,
<em><a href="https://bioconductor.org/packages/3.20/ENmix" class="external-link">ENmix</a></em>, and
<em><a href="https://bioconductor.org/packages/3.20/wateRmelon" class="external-link">wateRmelon</a></em>.
Additional functionality is provided via the <code>ewastools</code>
package, which is installed from GitHub using
<code>devtools::install_github("hhhh5/ewastools")</code>. Downstream
statistical modelling relies on base R and CRAN frameworks, including
generalised linear models and linear mixed-effects models. Users are
expected to have basic familiarity with R, command-line execution, and
Illumina IDAT file structures. Downstream statistical modelling relies
on base R and CRAN frameworks, including generalised linear models and
linear mixed-effects models. Users are expected to have basic
familiarity with R, command-line execution, and Illumina IDAT file
structures.</p>
<p>If you are asking yourself the question “Where do I start using
Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU" class="external-link">this
blog post</a>.</p>
</div>
<div class="section level3">
<h3 id="asking-for-help">Asking for help<a class="anchor" aria-label="anchor" href="#asking-for-help"></a>
</h3>
<p>As package developers, we try to explain clearly how to use our
packages and in which order to use the functions. But <code>R</code> and
<code>Bioconductor</code> have a steep learning curve so it is critical
to learn where to ask for help. The blog post quoted above mentions some
but we would like to highlight the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a>
as the main resource for getting help: remember to use the
<code>dnaEPICO</code> tag and check <a href="https://support.bioconductor.org/tag/dnaEPICO/" class="external-link">the older
posts</a>. Other alternatives are available such as creating GitHub
issues and tweeting. However, please note that if you want to receive
help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting
guidelines</a>. It is particularly critical that you provide a small
reproducible example and your session information so package developers
can track down the source of the error.</p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-minfi" class="csl-entry">
Aryee, Martin J, Andrew E Jaffe, Hector Corrada Bravo, Christine
Ladd-Acosta, Andrew P Feinberg, Kasper D Hansen, and Rafael A Irizarry.
2014. <span>“<span class="nocase">Minfi: a flexible and comprehensive
Bioconductor package for the analysis of Infinium DNA methylation
microarrays</span>.”</span> <em>Bioinformatics</em> 30 (10): 1363–69. <a href="https://doi.org/10.1093/bioinformatics/btu049" class="external-link">https://doi.org/10.1093/bioinformatics/btu049</a>.
</div>
<div id="ref-lme4" class="csl-entry">
Bates, Douglas, Martin Mächler, Ben Bolker, and Steve Walker. 2015.
<span>“Fitting Linear Mixed-Effects Models Using <span class="nocase">lme4</span>.”</span> <em>Journal of Statistical
Software</em> 67 (1): 1–48. <a href="https://doi.org/10.18637/jss.v067.i01" class="external-link">https://doi.org/10.18637/jss.v067.i01</a>.
</div>
<div id="ref-funnorm" class="csl-entry">
Fortin, Jean-Philippe, Aurélie Labbe, Mathieu Lemire, Brent W Zanke,
Thomas J Hudson, Elana J Frtig, Celia MT Greenwood, and Kasper D Hansen.
2014. <span>“<span class="nocase">Functional normalization of 450k
methylation array data improves replication in large cancer
studies</span>.”</span> <em>Genome Biology</em> 15 (11): 503. <a href="https://doi.org/10.1186/s13059-014-0503-2" class="external-link">https://doi.org/10.1186/s13059-014-0503-2</a>.
</div>
<div id="ref-minfiEPIC" class="csl-entry">
Fortin, Jean-Philippe, Timothy Triche Jr., and Kasper D Hansen. 2017.
<span>“Preprocessing, Normalization and Integration of the
<span>Illumina</span> <span>HumanMethylationEPIC</span> Array with
Minfi.”</span> <em>Bioinformatics</em> 33 (4): 558–60. <a href="https://doi.org/10.1093/bioinformatics/btw691" class="external-link">https://doi.org/10.1093/bioinformatics/btw691</a>.
</div>
<div id="ref-SWAN" class="csl-entry">
Maksimovic, Jovana, Lavinia Gordon, and Alicia Oshlack. 2012.
<span>“<span class="nocase">SWAN: Subset quantile Within-Array
Normalization for Illumina Infinium HumanMethylation450
BeadChips</span>.”</span> <em>Genome Biology</em> 13 (6): R44. <a href="https://doi.org/10.1186/gb-2012-13-6-r44" class="external-link">https://doi.org/10.1186/gb-2012-13-6-r44</a>.
</div>
<div id="ref-glm2" class="csl-entry">
Marschner, Ian. 2011. <span>“Glm2: Fitting Generalized Linear Models
with Convergence Problems.”</span> <em>The R Journal</em> 3: 12–15. <a href="https://doi.org/10.32614/RJ-2011-012" class="external-link">https://doi.org/10.32614/RJ-2011-012</a>.
</div>
<div id="ref-ewastools" class="csl-entry">
Murat, Kubra, Björn Grüning, Pawel W Poterlowicz, Gareth Westgate,
Desmond J Tobin, and Krzysztof Poterlowicz. 2020. <span>“<span class="nocase">Ewastools: Infinium Human Methylation BeadChip pipeline
for population epigenetics integrated into Galaxy</span>.”</span>
<em>GigaScience</em> 9 (5): giaa049. <a href="https://doi.org/10.1093/gigascience/giaa049" class="external-link">https://doi.org/10.1093/gigascience/giaa049</a>.
</div>
<div id="ref-wateRmelon" class="csl-entry">
Pidsley, Ruth, Ching Ching Y Wong, Matteo Volta, Katie Lunnon, Jonathan
Mill, and Leonard C Schalkwyk. 2013. <span>“<span class="nocase">A
data-driven approach to preprocessing Illumina 450K methylation array
data</span>.”</span> <em>BMC Genomics</em> 14: 293. <a href="https://doi.org/10.1186/1471-2164-14-293" class="external-link">https://doi.org/10.1186/1471-2164-14-293</a>.
</div>
<div id="ref-quantile" class="csl-entry">
Touleimat, Nizar, and Jörg Tost. 2012. <span>“Complete Pipeline for
Infinium() <span>Human</span> <span>Methylation</span> <span>450K</span>
<span>BeadChip</span> Data Processing Using Subset Quantile
Normalization for Accurate <span>DNA</span> Methylation
Estimation.”</span> <em>Epigenomics</em> 4 (3): 325–41. <a href="https://doi.org/10.2217/epi.12.21" class="external-link">https://doi.org/10.2217/epi.12.21</a>.
</div>
<div id="ref-noob" class="csl-entry">
Triche, Timothy J, Daniel J Weisenberger, David Van Den Berg, Peter W
Laird, and Kimberly D Siegmund. 2013. <span>“<span class="nocase">Low-level processing of Illumina Infinium DNA Methylation
BeadArrays.</span>”</span> <em>Nucleic Acids Research</em> 41 (7): e90.
<a href="https://doi.org/10.1093/nar/gkt090" class="external-link">https://doi.org/10.1093/nar/gkt090</a>.
</div>
<div id="ref-ENmix" class="csl-entry">
Xu, Zongli, Li Niu, and Jack A Taylor. 2021. <span>“<span class="nocase">The ENmix DNA methylation analysis pipeline for Illumina
BeadChip and comparisons with seven other preprocessing
pipelines</span>.”</span> <em>Clinical Epigenetics</em> 13 (1): 216. <a href="https://doi.org/10.1186/s13148-021-01207-1" class="external-link">https://doi.org/10.1186/s13148-021-01207-1</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Paul Ruiz, Divya Metha.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
